<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />


<title>Differential Gene Expression</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.28/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MDA-MB-453_DHT-RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Workflow Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
    <li>
      <a href="dge_enrichment.html">Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Gene Expression</h1>
<h4 class="author">Stephen Pederson<br>Dame Roma Mitchell Cancer
Research Laboratories<br>Adelaide Medical School<br>University of
Adelaide</h4>
<h4 class="date">21 September, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>MDA-MB-453_DHT-RNASeq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLMDAMB453DHTRNASeqtree9aaff4f369f404afe341b2b97af711cb4f52dc88targetblank9aaff4fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/tree/9aaff4f369f404afe341b2b97af711cb4f52dc88" target="_blank">9aaff4f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLMDAMB453DHTRNASeqtree9aaff4f369f404afe341b2b97af711cb4f52dc88targetblank9aaff4fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/tree/9aaff4f369f404afe341b2b97af711cb4f52dc88" target="_blank">9aaff4f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/sPLS-DA.Rmd
    Untracked:  analysis/sSNAPPY.Rmd
    Untracked:  output/MDA453_permutedscore.rds

Unstaged changes:
    Modified:   analysis/dge_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/dge_analysis.Rmd</code>) and HTML
(<code>docs/dge_analysis.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/9aaff4f369f404afe341b2b97af711cb4f52dc88/docs/dge_analysis.html" target="_blank">9aaff4f</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/28421b79fa110a042a0a486dd6cf4aa09e059bb4/analysis/dge_analysis.Rmd" target="_blank">28421b7</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
<td>
Exported PCA as pdf
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/838f1b8754b7ac9bd880e7dd2921a65bbb8d1e99/docs/dge_analysis.html" target="_blank">838f1b8</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/745652668b05303a256e152a6d4c1dabe18c8fdd/analysis/dge_analysis.Rmd" target="_blank">7456526</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
<td>
Added pdf export &amp; fixed dodgy cpm bug
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/455e3bcf15b99512d3c229517eff372bc3eb0aa2/docs/dge_analysis.html" target="_blank">455e3bc</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/b14fd077facfd42f5eaba2dad15e5b81d3075f8f/analysis/dge_analysis.Rmd" target="_blank">b14fd07</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-03
</td>
<td>
Tidied up Rmd files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/cc3b130cbdd2157b425f68d2a47f24127097d870/docs/dge_analysis.html" target="_blank">cc3b130</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/b1df68ce5373a7929b0e561bf4d939d6fbd096dd/analysis/dge_analysis.Rmd" target="_blank">b1df68c</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-11-03
</td>
<td>
Added export of CPM values
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/1d06dff563725688789f0fb9c68f321c15663e79/docs/dge_analysis.html" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/f6ff160a227ae55293eaaf53b14a27897de11fa7/analysis/dge_analysis.Rmd" target="_blank">f6ff160</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
<td>
Copied across previous analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/MDA-MB-453_DHT-RNASeq/7d6936a3eeca8ca7f3fc3a1269377ad0db9305da/docs/dge_analysis.html" target="_blank">7d6936a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2020-10-13
</td>
<td>
Finished run
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/645309c43115ea0b7aceff4cc3b8b0e56a7730ba/analysis/dge_analysis.Rmd" target="_blank">645309c</a>
</td>
<td>
Steve Pederson
</td>
<td>
2020-10-13
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<pre class="r"><code>if (!dir.exists(here::here(&quot;docs/assets&quot;))) dir.create(here::here(&quot;docs/assets&quot;))</code></pre>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- read_rds(here::here(&quot;output/genesGR.rds&quot;))</code></pre>
<p>Gene annotations were again loaded from Ensembl Release 101. The <a
href="qc_aligned.html#Annotation_Setup">previously defined</a>
<code>GenomicRanges</code> object containing GC content and Gene Length
was also loaded, containing information for 67,990 genes.</p>
</div>
<div id="sample-merging" class="section level2">
<h2>Sample Merging</h2>
<pre class="r"><code>samples &lt;- config$samples %&gt;%
  here::here() %&gt;%
  read_tsv() %&gt;%
  mutate(
    Filename = paste0(sample, suffix)
  ) %&gt;%
  dplyr::select(Filename, flowcell_id, treat, rep, desc) %&gt;%
  mutate(name = str_extract(desc, &quot;MDA[0-9].++&quot;)) %&gt;%
  dplyr::select(-desc) %&gt;%
  mutate_if(
    function(x){length(unique(x)) &lt; length(x)},
    as.factor
  ) %&gt;%
  mutate(
    treat = relevel(treat, ref = &quot;Vehicle&quot;)
  )</code></pre>
<pre class="r"><code>mergedSamples &lt;- samples %&gt;%
  group_by(name, rep, treat) %&gt;%
  tally()</code></pre>
<pre class="r"><code>pander(
  mergedSamples,
  caption = &quot;*Summary of sequencing total runs for all samples*&quot;
)</code></pre>
<table style="width:40%;">
<caption><em>Summary of sequencing total runs for all
samples</em></caption>
<colgroup>
<col width="12%" />
<col width="8%" />
<col width="13%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">name</th>
<th align="center">rep</th>
<th align="center">treat</th>
<th align="center">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">MDA1-D</td>
<td align="center">1</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA1-V</td>
<td align="center">1</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
<tr class="odd">
<td align="center">MDA2-D</td>
<td align="center">2</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA2-V</td>
<td align="center">2</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
<tr class="odd">
<td align="center">MDA3-D</td>
<td align="center">3</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA3-V</td>
<td align="center">3</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
<tr class="odd">
<td align="center">MDA4-D</td>
<td align="center">4</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA4-V</td>
<td align="center">4</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
<tr class="odd">
<td align="center">MDA5-D</td>
<td align="center">5</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA5-V</td>
<td align="center">5</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
<tr class="odd">
<td align="center">MDA6-D</td>
<td align="center">6</td>
<td align="center">DHT</td>
<td align="center">8</td>
</tr>
<tr class="even">
<td align="center">MDA6-V</td>
<td align="center">6</td>
<td align="center">Vehicle</td>
<td align="center">8</td>
</tr>
</tbody>
</table>
<p>This data set is slightly unique in that samples were sequenced
across multiple lanes and flowcells. Metadata was re-organised to enable
summarising of counts across the multiple sequencing runs obtained for
every sample. The <a href="qc_aligned.html#pca">initial PCA</a> showed
that variability between replicates and flowcells was dwarfed by
variability between samples. As such, all counts were able to be merged
to give a single set of counts for each individual sample.</p>
<pre class="r"><code>treat_cols &lt;- hcl.colors(
  n = length(levels(mergedSamples$treat)), 
  palette = &quot;Red-Blue&quot;, 
  rev = TRUE
  ) %&gt;%
  setNames(levels(mergedSamples$treat)) 
rep_cols &lt;- hcl.colors(
  n = length(levels(mergedSamples$rep)), 
  palette = &quot;Spectral&quot;
  ) %&gt;%
  setNames(levels(mergedSamples$rep))</code></pre>
</div>
<div id="count-data" class="section level2">
<h2>Count Data</h2>
<pre class="r"><code>counts &lt;- here::here(&quot;data/aligned/counts/counts.out&quot;) %&gt;%
  read_tsv(comment = &quot;#&quot;) %&gt;%
  dplyr::select(Geneid, ends_with(&quot;bam&quot;)) %&gt;%
  rename_at(vars(ends_with(&quot;bam&quot;)), dirname) %&gt;%
  rename_all(basename) %&gt;%
  column_to_rownames(&quot;Geneid&quot;)</code></pre>
<pre class="r"><code>mergedCounts &lt;- counts %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  pivot_longer(
    cols = -gene_id,
    names_to = &quot;Filename&quot;,
    values_to = &quot;counts&quot;
  ) %&gt;%
  left_join(samples) %&gt;%
  group_by(
    gene_id, name, rep, treat
  ) %&gt;%
  summarise(counts = sum(counts), .groups = &quot;drop&quot;) %&gt;%
  pivot_wider(
    id_cols = gene_id,
    values_from = counts,
    names_from = name
  ) %&gt;%
  column_to_rownames(&quot;gene_id&quot;)</code></pre>
<pre class="r"><code>minCPM &lt;- 1.5
minSamples &lt;- ncol(mergedCounts) / 2
genes2Keep &lt;- mergedCounts %&gt;%
  cpm() %&gt;%
  is_greater_than(minCPM) %&gt;%
  rowSums() %&gt;%
  is_weakly_greater_than(minSamples) </code></pre>
<p>The criteria for a gene to be considered as detected, <span
class="math inline">\(&gt;\)</span> 1.5 counts per million (CPM) were
required to observed in <span class="math inline">\(\geq\)</span> 6
samples. This effectively ensured <span
class="math inline">\(\geq\)</span> 5 counts in at least 6 merged
libraries.</p>
<p>Of the 60,671 genes contained in the annotation for this release,
46,112 genes were removed as failing this criteria for detection,
leaving 14,559 genes for downstream analysis.</p>
<pre class="r"><code>a &lt;- mergedCounts %&gt;%
  cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = contains(&quot;MDA&quot;),
    names_to = &quot;name&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(samples) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = rep, linetype = treat)) +
  geom_density() +
  scale_colour_manual(values= rep_cols) +
  labs(
    y = &quot;Density&quot;,
    colour = &quot;Replicate&quot;,
    linetype = &quot;Treatment&quot;
  )
b &lt;- mergedCounts[genes2Keep,] %&gt;%
  cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = contains(&quot;MDA&quot;),
    names_to = &quot;name&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(samples) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = rep, linetype = treat)) +
  geom_density() +
  scale_colour_manual(values= rep_cols) +
  labs(
    y = &quot;Density&quot;,
    colour = &quot;Replicate&quot;,
    linetype = &quot;Treatment&quot;
  )
plot_grid(
  a + theme(legend.position = &quot;none&quot;), 
  b + theme(legend.position = &quot;none&quot;), 
  get_legend(a),
  labels = c(&quot;A&quot;, &quot;B&quot;),
  rel_widths = c(4, 4, 1),
  nrow = 1
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotDensities-1.png" alt="*Distributions of logCPM values on merged counts, A) before and B) after filtering of undetectable genes. Some differences between replicates were noted.*" width="960" />
<p class="caption">
<em>Distributions of logCPM values on merged counts, A) before and B)
after filtering of undetectable genes. Some differences between
replicates were noted.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotDensities-1">
Past versions of plotDensities-1.png
</button>
</p>
<div id="fig-plotDensities-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotDensities-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dge &lt;- mergedCounts[genes2Keep,] %&gt;%
  DGEList(
    samples = mergedSamples %&gt;%
      as.data.frame() %&gt;%
      set_rownames(.$name) %&gt;%
      .[,colnames(.)],
    group = mergedSamples %&gt;%
      as.data.frame() %&gt;%
      set_rownames(.$name) %&gt;%
      .[,colnames(.)] %&gt;%
      pull(treat),
    genes = genesGR[rownames(.)]
  ) %&gt;%
  calcNormFactors()</code></pre>
</div>
</div>
<div id="qc" class="section level1">
<h1>QC</h1>
<div id="library-sizes" class="section level2">
<h2>Library Sizes</h2>
<pre class="r"><code>dge$samples %&gt;%
  ggplot(aes(name, lib.size, fill = treat)) +
  geom_col() +
  geom_hline(yintercept = 1e7, linetype = 2) +
  facet_wrap(~treat, scales = &quot;free_x&quot;) +
  scale_y_continuous(
    labels = comma, expand = expansion(c(0, 0.05))
  ) +
  scale_fill_manual(values= treat_cols) +
  labs(x = &quot;Sample Name&quot;, y = &quot;Library Size&quot;) +
  theme(legend.position = &quot;none&quot;) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotLibSizes-1.png" alt="*Library sizes after removal of undetectable genes. The common-use minimum library size of 10 million reads is shown as a dashed line.*" width="960" />
<p class="caption">
<em>Library sizes after removal of undetectable genes. The common-use
minimum library size of 10 million reads is shown as a dashed line.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLibSizes-1">
Past versions of plotLibSizes-1.png
</button>
</p>
<div id="fig-plotLibSizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotLibSizes-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Given that previous work had only assessed the libraries prior to
merging, the library sizes were checked after merging replicate
sequencing runs. The median library size was found to be 13.6 million
reads which is just above the common-use minimum recommendation of 10
million reads/sample.</p>
<p>However, sample MDA6 was below the common-use minimum library size in
both treatment groups and whilst discarding this sample is not advised,
this should be noted for all downstream analysis. Similarly, the
DHT-treated sample for MDA2 showed a library size which was nearly
double the rest of the samples</p>
</div>
<div id="pca-pre-normalisation" class="section level2">
<h2>PCA: Pre Normalisation</h2>
<pre class="r"><code>pca &lt;- dge %&gt;%
  cpm(log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
<pre class="r"><code>a &lt;- pca %&gt;%
  autoplot(
    data = dge$samples, 
    x = 1, y = 3,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
b &lt;- pca %&gt;%
  autoplot(
    data = dge$samples, 
    x = 2, y = 3,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
c &lt;- pca %&gt;%
  autoplot(
    data = dge$samples, 
    x = 1, y = 2,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  geom_segment(
    aes(
      x = Vehicle_PC1, xend = DHT_PC1, 
      y = Vehicle_PC2, yend = DHT_PC2,
      colour = rep
    ),
    data = . %&gt;%
      dplyr::select(any_of(colnames(dge$samples)), PC1, PC2) %&gt;% 
      pivot_longer(cols = starts_with(&quot;PC&quot;), names_to = &quot;PC&quot;, values_to = &quot;value&quot;) %&gt;% 
      pivot_wider(names_from = c(&quot;treat&quot;, &quot;PC&quot;), values_from = value, id_cols = c(&quot;rep&quot;)),
    alpha = 0.4,
    arrow = arrow()
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
plot_grid(
  a + theme(legend.position = &quot;none&quot;), 
  b + theme(legend.position = &quot;none&quot;),
  c + theme(legend.position = &quot;none&quot;),
  get_legend(a),
  labels = LETTERS[1:3],
  nrow = 2
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotPCA-1.png" alt="*PCA on logCPM from merged counts. Replicate 2 appears problematic in both treatments, whilst the Vehicle samples from replicate 5 also appears slightly divergent from other vehicle samples. PC1 appears to capture a non-experimental artefact whilst PC2 captures most of the intended biology, as shown by the arrows which move from Vehicle to DHT treated samples.*" width="768" />
<p class="caption">
<em>PCA on logCPM from merged counts. Replicate 2 appears problematic in
both treatments, whilst the Vehicle samples from replicate 5 also
appears slightly divergent from other vehicle samples. PC1 appears to
capture a non-experimental artefact whilst PC2 captures most of the
intended biology, as shown by the arrows which move from Vehicle to DHT
treated samples.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA-1">
Past versions of plotPCA-1.png
</button>
</p>
<div id="fig-plotPCA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotPCA-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Despite concerns raised by the library sizes, sample MDA6 appears to
be relatively consistent with other samples of the same treatment group.
However, MDA2 showed significant divergence from the others in both
treatment groups, whilst the vehicle-treated (i.e. control) sample from
MDA5 was also a notable outlier.</p>
</div>
<div id="checks-for-gc-and-length-bias" class="section level2">
<h2>Checks for GC and Length bias</h2>
<p>Given the clear unexpected variability captured by PC1, the impact of
GC content or length bias was assessed as a possible source.</p>
<p>Genes were divided in 10 approximately equal sized bins based on
increasing length, and 10 approximately equal sized bins based on
increasing GC content, with the final GC/Length bins being the
combination 100 bins using both sets. The contribution of each gene to
PC1 and PC2 was assessed and a t-test performed on each bin. This
tests</p>
<p><span class="math display">\[
H_0: \mu = 0 \text{ against } H_A: \mu \neq 0
\]</span></p>
<p>where <span class="math inline">\(\mu\)</span> represents the true
contribution to PC1 of all genes in that bin.</p>
<p>If any bin makes a contribution to PC1 the mean will be clearly
non-zero, whilst if there is no contribution the mean will be near zero.
In this way, the impact of gene length and GC content on variance within
the dataset can be assessed. As seen below, the contribution of GC
content and gene length to PC1 is very clear, with a smaller
contribution being evident across PC2. As a result, Conditional Quantile
Normalisation (<code>CQN</code>) is recommended in preference to the
more common TMM normalisation.</p>
<pre class="r"><code>dge$genes %&gt;%
  dplyr::select(gene_id, ave_tx_len, gc_content) %&gt;%
  mutate(
    GC = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    Length = cut(
      x = ave_tx_len,
      labels = seq_len(10),
      breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(GC, Length, sep = &quot;_&quot;),
    PC1 = pca$rotation[gene_id, &quot;PC1&quot;],
    PC2 = pca$rotation[gene_id, &quot;PC2&quot;]
  ) %&gt;%
  pivot_longer(
    cols = c(&quot;PC1&quot;, &quot;PC2&quot;),
    names_to = &quot;PC&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  group_by(PC, GC, Length, bin) %&gt;%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    adjP = p.adjust(p, method = &quot;bonf&quot;)
  ) %&gt;%
  ggplot(
    aes(Length, GC, colour = t, alpha = -log10(adjP), size = Size)
  ) +
  geom_point() +
  facet_wrap(~PC) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  labs(alpha = expression(paste(-log[10], p))) +
  theme(
    panel.grid = element_blank(),
    legend.position = &quot;bottom&quot;
  ) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotPCA_GC_Length-1.png" alt="*Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values. The number of genes in each bin is indicated by the circle size. The clear pattern across PC1 is unambiguous.*" width="960" />
<p class="caption">
<em>Contribution of each GC/Length Bin to PC1 and PC2. Fill colours
indicate the t-statistic, with tranparency denoting significance as
-log10(p), using Bonferroni-adjusted p-values. The number of genes in
each bin is indicated by the circle size. The clear pattern across PC1
is unambiguous.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA_GC_Length-1">
Past versions of plotPCA_GC_Length-1.png
</button>
</p>
<div id="fig-plotPCA_GC_Length-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotPCA_GC_Length-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Given these results, the considered options for analysis are:</p>
<ul>
<li>Use CQN normalisation and GLM-QL Fits</li>
<li>Use limma/voom to take advantage of the ability to down-weight
samples</li>
<li>Exclude replicate 2, in combination with either of the above
strategies</li>
</ul>
</div>
<div id="normalisation" class="section level2">
<h2>Normalisation</h2>
<pre class="r"><code>cqNorm &lt;- with(
  dge,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$ave_tx_len
  )
)
dge$offset &lt;- cqNorm$glm.offset
logCPM &lt;- cqNorm$y + cqNorm$offset</code></pre>
<pre class="r"><code>a &lt;- cqNorm$func1 %&gt;%
  as.data.frame() %&gt;%
  mutate(x = cqNorm$grid1) %&gt;%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = &quot;name&quot;,
    values_to = &quot;QR fit&quot;
  ) %&gt;%
  left_join(dge$samples) %&gt;%
  ggplot(
    aes(x, `QR fit`, colour = rep, group = name, linetype = treat)
  ) +
  geom_line() +
  labs(x = &quot;GC content&quot;, colour = &quot;Replicate&quot;, linetype = &quot;Treatment&quot;)
b &lt;- cqNorm$func2 %&gt;%
  as.data.frame() %&gt;%
  mutate(x = cqNorm$grid2) %&gt;%
  pivot_longer(
    cols = any_of(colnames(dge)),
    names_to = &quot;name&quot;,
    values_to = &quot;QR fit&quot;
  ) %&gt;%
  left_join(dge$samples) %&gt;%
  ggplot(
    aes(x, `QR fit`, colour = rep, group = name, linetype = treat)
  ) +
  geom_line() +
  labs(
    x = expression(paste(log[10], &quot; Gene Length (kb)&quot;)),
    colour = &quot;Replicate&quot;, linetype = &quot;Treatment&quot;
  )
plot_grid(
  a + theme(legend.position = &quot;none&quot;), 
  b + theme(legend.position = &quot;none&quot;),
  get_legend(a),
  nrow = 1,
  rel_widths = c(3, 3, 1)
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotCQN-1.png" alt="*Model fits used when applying CQN. The divergent samples previously noted on the PCA are again quite divergent here. In particular, the long genes with low GC content appear to be where the primary differences are found, in keeping with the previous PCA analysis.*" width="960" />
<p class="caption">
<em>Model fits used when applying CQN. The divergent samples previously
noted on the PCA are again quite divergent here. In particular, the long
genes with low GC content appear to be where the primary differences are
found, in keeping with the previous PCA analysis.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotCQN-1">
Past versions of plotCQN-1.png
</button>
</p>
<div id="fig-plotCQN-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotCQN-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pcaPost &lt;- logCPM %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
<pre class="r"><code>a &lt;- pcaPost %&gt;%
  autoplot(
    data = dge$samples, 
    x = 1, y = 3,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
b &lt;- pcaPost %&gt;%
  autoplot(
    data = dge$samples, 
    x = 2, y = 3,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
c &lt;- pcaPost %&gt;%
  autoplot(
    data = dge$samples, 
    x = 1, y = 2,
    colour = &quot;rep&quot;, 
    shape = &quot;treat&quot;, 
    size = 3
  ) +
  geom_segment(
    aes(
      x = Vehicle_PC1, xend = DHT_PC1, 
      y = Vehicle_PC2, yend = DHT_PC2,
      colour = rep
    ),
    data = . %&gt;%
      dplyr::select(any_of(colnames(dge$samples)), PC1, PC2) %&gt;% 
      pivot_longer(cols = starts_with(&quot;PC&quot;), names_to = &quot;PC&quot;, values_to = &quot;value&quot;) %&gt;% 
      pivot_wider(names_from = c(&quot;treat&quot;, &quot;PC&quot;), values_from = value, id_cols = c(&quot;rep&quot;)),
    alpha = 0.4,
    arrow = arrow(),
    show.legend = FALSE
  ) +
  scale_colour_manual(values= rep_cols) +
  labs(
    colour = &quot;Replicate&quot;,
    shape = &quot;Treatment&quot;
  )
plot_grid(
  a + theme(legend.position = &quot;none&quot;), 
  b + theme(legend.position = &quot;none&quot;),
  c + theme(legend.position = &quot;none&quot;),
  get_legend(a),
  labels = LETTERS[1:3],
  nrow = 2
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotPcaPost-1.png" alt="*PCA on logCPM after performing CQN. Much of the variability previously seen in with replicate 5 has now been minimised, whilst replicate 2 still appears to be a considerable outlier. However, PC1 and PC2 now capture the biology in question, with vectors denoting the shift from vehicle to treatment shown as arrows shown in panel C. Panel B does highlight that along PC2 the control sample from replicate 2 does more closely resemble the treated samples from all other replicates.*" width="768" />
<p class="caption">
<em>PCA on logCPM after performing CQN. Much of the variability
previously seen in with replicate 5 has now been minimised, whilst
replicate 2 still appears to be a considerable outlier. However, PC1 and
PC2 now capture the biology in question, with vectors denoting the shift
from vehicle to treatment shown as arrows shown in panel C. Panel B does
highlight that along PC2 the control sample from replicate 2 does more
closely resemble the treated samples from all other replicates.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPcaPost-1">
Past versions of plotPcaPost-1.png
</button>
</p>
<div id="fig-plotPcaPost-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotPcaPost-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pdf(
  here::here(&quot;docs/assets/pcaPost.pdf&quot;), width = 8, height = 8
)
c
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>dge$genes %&gt;%
  dplyr::select(gene_id, ave_tx_len, gc_content) %&gt;%
  mutate(
    GC = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    Length = cut(
      x = ave_tx_len,
      labels = seq_len(10),
      breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(GC, Length, sep = &quot;_&quot;),
    `pre-CQN` = pca$rotation[gene_id, &quot;PC1&quot;],
    `post-CQN` = pcaPost$rotation[gene_id, &quot;PC1&quot;]
  ) %&gt;%
  pivot_longer(
    cols = c(&quot;pre-CQN&quot;, &quot;post-CQN&quot;),
    names_to = &quot;status&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  mutate(status = factor(status, levels = c(&quot;pre-CQN&quot;, &quot;post-CQN&quot;))) %&gt;%
  group_by(status, GC, Length, bin) %&gt;%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    FDR = p.adjust(p, method = &quot;bonf&quot;)
  ) %&gt;%
  ggplot(
    aes(Length, GC, colour = t, alpha = -log10(FDR), size = Size)
  ) +
  geom_point() +
  facet_wrap(~status) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  theme(
    panel.grid = element_blank(),
    legend.position = &quot;bottom&quot;
  ) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotPCA_GC_post-1.png" alt="*Contributions of gene length and GC content to PC1 shown both before and after CQN. The strong patterns seen before normalisation have clearly been reduced. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values. The number of genes in each bin is indicated by the circle size. The clear pattern across PC1 is unambiguous.*" width="960" />
<p class="caption">
<em>Contributions of gene length and GC content to PC1 shown both before
and after CQN. The strong patterns seen before normalisation have
clearly been reduced. Fill colours indicate the t-statistic, with
tranparency denoting significance as -log10(p), using
Bonferroni-adjusted p-values. The number of genes in each bin is
indicated by the circle size. The clear pattern across PC1 is
unambiguous.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA_GC_post-1">
Past versions of plotPCA_GC_post-1.png
</button>
</p>
<div id="fig-plotPCA_GC_post-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotPCA_GC_post-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>CQN clearly had a positive impact on the data, however, the exclusion
of replicate 2 may still be an option. Given the consistency seen with
the post-CQN PCA, using a blocking approach will also be the preferred
analytic approach.</p>
</div>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential Expression Analysis</h1>
<div id="main-results" class="section level2">
<h2>Main Results</h2>
<p>A design matrix was formed giving each replicate it’s own baseline
and the common DHT response was then specified as the final column.
Following this, dispersions were estimated for the main
<code>DGEList</code> object and the model fitted using the
Quasi-likelihood GLM.</p>
<pre class="r"><code>X &lt;- model.matrix(~0 + rep + treat, data = dge$samples) %&gt;%
  set_colnames(str_remove(colnames(.), &quot;treat&quot;))
dge &lt;- estimateDisp(dge, design = X, robust = TRUE)
fit &lt;- glmQLFit(dge)</code></pre>
<pre class="r"><code>alpha &lt;- 0.05
lambda &lt;- log2(1.2)
topTable &lt;- glmTreat(fit, coef = &quot;DHT&quot;, lfc = lambda) %&gt;%
  topTags(n = Inf) %&gt;%
  .[[&quot;table&quot;]] %&gt;%
  as_tibble() %&gt;%
  mutate(
    location = paste0(seqnames, &quot;:&quot;, start, &quot;-&quot;, end, &quot;:&quot;, strand),
    rankingStat = -sign(logFC)*log10(PValue),
    signedRank = rank(rankingStat),
    DE = FDR &lt; alpha
  ) %&gt;%
  dplyr::select(
    gene_id, gene_name, logCPM, logFC, PValue, FDR, 
    location, gene_biotype, entrezid, ave_tx_len, gc_content, 
    rankingStat, signedRank, DE
  )
de &lt;- dplyr::filter(topTable, DE)$gene_id
up &lt;- dplyr::filter(topTable, DE, logFC &gt; 0)$gene_id
down &lt;- dplyr::filter(topTable, DE, logFC &lt; 0)$gene_id</code></pre>
<p>In order to detect genes which respond to DHT the Null Hypothesis
(<span class="math inline">\(H_0\)</span>) was specified to be a range
around zero, instead of the conventional point-value.</p>
<p><span class="math display">\[
H_0: -\lambda \leq \mu \leq \lambda \\
\text{Vs.} \\
H_A: |\mu| &gt; \lambda
\]</span></p>
<p>This characterises <span class="math inline">\(H_0\)</span> as being
a range instead of being the singularity 0, with <span
class="math inline">\(\mu\)</span> representing the true mean logFC. The
default value of <span class="math inline">\(\lambda = \log_2 1.2 =
0.263\)</span> was chosen. This removes any requirement for <em>post
hoc</em> filtering based on logFC. Using this approach, 619 genes were
considered as DE to an FDR of 0.05. Of these, 387 were up-regulated with
the remaining 232 being down-regulated.</p>
<pre class="r"><code>topTable %&gt;%
  ggplot(aes(logCPM, logFC)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %&gt;%
      dplyr::filter(logFC &gt; 2 | logFC &lt; -2),
    show.legend = FALSE
  ) +
  geom_smooth(se = FALSE) +
  scale_colour_manual(
    values = c(&quot;grey50&quot;, &quot;red&quot;)
  ) +
  theme(
    legend.position = &quot;none&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotMA-1.png" alt="*MA plot with blue curve through the data showing the GAM fit. The curve for low-expressed genes indicated a potential bias remains in the data, whilst this is not evident for highly expressed genes. DE genes are highlighted in red.*" width="960" />
<p class="caption">
<em>MA plot with blue curve through the data showing the GAM fit. The
curve for low-expressed genes indicated a potential bias remains in the
data, whilst this is not evident for highly expressed genes. DE genes
are highlighted in red.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotMA-1">
Past versions of plotMA-1.png
</button>
</p>
<div id="fig-plotMA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/838f1b8754b7ac9bd880e7dd2921a65bbb8d1e99/docs/figure/dge_analysis.Rmd/plotMA-1.png" target="_blank">838f1b8</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotMA-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(
  here::here(&quot;docs/assets/plotMA.pdf&quot;),
  width = 10, height = 6,
  units = &quot;in&quot;
)</code></pre>
<pre class="r"><code>topTable %&gt;%
  ggplot(aes(logFC, -log10(PValue))) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_text_repel(
    aes(label = gene_name, colour = DE),
    data = . %&gt;%
      dplyr::filter(abs(logFC) &gt; 2 | PValue &lt; 5e-10),
    show.legend = FALSE
  ) +
  geom_vline(xintercept = c(-1,1), colour = &quot;blue&quot;, linetype = 2) +
  labs(
    y = expression(paste(-log[10], &quot;p&quot;))
  ) +
  scale_colour_manual(
    values = c(&quot;grey50&quot;, &quot;red&quot;)
  ) +
  theme(
    legend.position = &quot;none&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotVolcano-1.png" alt="*Volcano plot showing significance against log fold-change. The common-use cutoff values for considering a gene as DE (|logFC| &gt; 1) are shown in blue as a guide only as these were not used for this analysis.*" width="960" />
<p class="caption">
<em>Volcano plot showing significance against log fold-change. The
common-use cutoff values for considering a gene as DE (|logFC| &gt; 1)
are shown in blue as a guide only as these were not used for this
analysis.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotVolcano-1">
Past versions of plotVolcano-1.png
</button>
</p>
<div id="fig-plotVolcano-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/838f1b8754b7ac9bd880e7dd2921a65bbb8d1e99/docs/figure/dge_analysis.Rmd/plotVolcano-1.png" target="_blank">838f1b8</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotVolcano-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(
  here::here(&quot;docs/assets/plotVolcano.pdf&quot;),
  width = 10, height = 8,
  units = &quot;in&quot;
)</code></pre>
<pre class="r"><code>topTable %&gt;%
  dplyr::filter(DE) %&gt;%
  dplyr::select(-DE, -gc_content, -ave_tx_len, -contains(&quot;rank&quot;)) %&gt;%
  mutate(
    across(c(&quot;logFC&quot;, &quot;logCPM&quot;), round, digits = 2)
  ) %&gt;%
  datatable(
    rownames = FALSE, 
    caption = glue(&quot;*All genes were considered as DE using the criteria of an FDR-adjusted p-value &lt; {alpha}.*&quot;)
  ) %&gt;%
  formatSignif(
    columns = c(&quot;PValue&quot;, &quot;FDR&quot;),
    digits = 3
  )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4143bfbf22be9ab3c92e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4143bfbf22be9ab3c92e">{"x":{"filter":"none","vertical":false,"caption":"<caption>*All genes were considered as DE using the criteria of an FDR-adjusted p-value &lt; 0.05.*<\/caption>","data":[["ENSG00000100003","ENSG00000165272","ENSG00000182795","ENSG00000164188","ENSG00000100906","ENSG00000186526","ENSG00000096060","ENSG00000128641","ENSG00000117009","ENSG00000136997","ENSG00000047644","ENSG00000143507","ENSG00000156284","ENSG00000213759","ENSG00000188064","ENSG00000174469","ENSG00000111666","ENSG00000177508","ENSG00000151790","ENSG00000079691","ENSG00000127663","ENSG00000135226","ENSG00000101298","ENSG00000241878","ENSG00000108932","ENSG00000152620","ENSG00000073282","ENSG00000175264","ENSG00000179841","ENSG00000169116","ENSG00000158125","ENSG00000141384","ENSG00000176387","ENSG00000176597","ENSG00000166068","ENSG00000085377","ENSG00000168502","ENSG00000110104","ENSG00000146535","ENSG00000133424","ENSG00000113645","ENSG00000139211","ENSG00000154734","ENSG00000005469","ENSG00000181418","ENSG00000184588","ENSG00000150594","ENSG00000160862","ENSG00000049759","ENSG00000169083","ENSG00000124224","ENSG00000198435","ENSG00000058600","ENSG00000166197","ENSG00000145569","ENSG00000214313","ENSG00000145703","ENSG00000165732","ENSG00000132382","ENSG00000017797","ENSG00000135899","ENSG00000142197","ENSG00000179104","ENSG00000134516","ENSG00000137709","ENSG00000164823","ENSG00000249956","ENSG00000107815","ENSG00000101846","ENSG00000188001","ENSG00000158050","ENSG00000167183","ENSG00000166387","ENSG00000109670","ENSG00000138829","ENSG00000275342","ENSG00000163884","ENSG00000143153","ENSG00000147689","ENSG00000130826","ENSG00000171617","ENSG00000157911","ENSG00000065183","ENSG00000121966","ENSG00000103460","ENSG00000204103","ENSG00000069974","ENSG00000120899","ENSG00000106541","ENSG00000130810","ENSG00000099860","ENSG00000179041","ENSG00000162817","ENSG00000124357","ENSG00000048162","ENSG00000164574","ENSG00000151835","ENSG00000155380","ENSG00000255774","ENSG00000144218","ENSG00000116337","ENSG00000250488","ENSG00000118965","ENSG00000137460","ENSG00000177398","ENSG00000151715","ENSG00000215110","ENSG00000167291","ENSG00000110660","ENSG00000078804","ENSG00000164938","ENSG00000261236","ENSG00000214252","ENSG00000274173","ENSG00000115758","ENSG00000071967","ENSG00000116133","ENSG00000121057","ENSG00000076716","ENSG00000121858","ENSG00000269516","ENSG00000280623","ENSG00000147642","ENSG00000047648","ENSG00000179094","ENSG00000144824","ENSG00000166833","ENSG00000145220","ENSG00000165271","ENSG00000196091","ENSG00000114767","ENSG00000150556","ENSG00000125629","ENSG00000137124","ENSG00000071282","ENSG00000156675","ENSG00000177192","ENSG00000170577","ENSG00000148702","ENSG00000182057","ENSG00000177606","ENSG00000135821","ENSG00000135373","ENSG00000104356","ENSG00000182957","ENSG00000155846","ENSG00000141738","ENSG00000132768","ENSG00000123572","ENSG00000120254","ENSG00000115556","ENSG00000157600","ENSG00000124762","ENSG00000115738","ENSG00000102780","ENSG00000162409","ENSG00000029725","ENSG00000198740","ENSG00000272622","ENSG00000112578","ENSG00000150687","ENSG00000112773","ENSG00000164442","ENSG00000163406","ENSG00000053108","ENSG00000104081","ENSG00000147100","ENSG00000232457","ENSG00000134323","ENSG00000124209","ENSG00000165689","ENSG00000131791","ENSG00000183111","ENSG00000154274","ENSG00000160145","ENSG00000145088","ENSG00000144815","ENSG00000151914","ENSG00000169750","ENSG00000113739","ENSG00000163811","ENSG00000146376","ENSG00000163659","ENSG00000110107","ENSG00000106571","ENSG00000170017","ENSG00000132437","ENSG00000081051","ENSG00000171004","ENSG00000071575","ENSG00000124784","ENSG00000157613","ENSG00000108244","ENSG00000106785","ENSG00000185133","ENSG00000067057","ENSG00000116717","ENSG00000116649","ENSG00000095585","ENSG00000170209","ENSG00000041353","ENSG00000128294","ENSG00000141736","ENSG00000130701","ENSG00000149218","ENSG00000177788","ENSG00000182704","ENSG00000154783","ENSG00000060971","ENSG00000261104","ENSG00000154153","ENSG00000111077","ENSG00000095739","ENSG00000116991","ENSG00000147224","ENSG00000198650","ENSG00000103222","ENSG00000068366","ENSG00000260228","ENSG00000125630","ENSG00000131844","ENSG00000123728","ENSG00000171444","ENSG00000134508","ENSG00000033170","ENSG00000117461","ENSG00000107731","ENSG00000204856","ENSG00000167721","ENSG00000064763","ENSG00000149292","ENSG00000204178","ENSG00000179431","ENSG00000148735","ENSG00000007545","ENSG00000136826","ENSG00000232677","ENSG00000197375","ENSG00000145860","ENSG00000162645","ENSG00000015133","ENSG00000082516","ENSG00000139146","ENSG00000105835","ENSG00000151726","ENSG00000261409","ENSG00000134574","ENSG00000163053","ENSG00000136141","ENSG00000035664","ENSG00000091127","ENSG00000158859","ENSG00000068615","ENSG00000164163","ENSG00000144840","ENSG00000176055","ENSG00000077348","ENSG00000041982","ENSG00000176641","ENSG00000127863","ENSG00000176842","ENSG00000148840","ENSG00000104369","ENSG00000115946","ENSG00000166446","ENSG00000136068","ENSG00000165912","ENSG00000139197","ENSG00000183569","ENSG00000242114","ENSG00000196756","ENSG00000125398","ENSG00000168118","ENSG00000247626","ENSG00000122176","ENSG00000124466","ENSG00000174705","ENSG00000233870","ENSG00000058668","ENSG00000140836","ENSG00000146197","ENSG00000151575","ENSG00000110328","ENSG00000168140","ENSG00000090487","ENSG00000165959","ENSG00000115963","ENSG00000161642","ENSG00000175602","ENSG00000086619","ENSG00000164543","ENSG00000181026","ENSG00000139116","ENSG00000196954","ENSG00000125508","ENSG00000145685","ENSG00000144452","ENSG00000116191","ENSG00000112208","ENSG00000259153","ENSG00000000457","ENSG00000142627","ENSG00000133316","ENSG00000123836","ENSG00000235713","ENSG00000117602","ENSG00000116141","ENSG00000189046","ENSG00000114790","ENSG00000113916","ENSG00000136718","ENSG00000079805","ENSG00000235092","ENSG00000064651","ENSG00000178607","ENSG00000180611","ENSG00000260573","ENSG00000158470","ENSG00000173145","ENSG00000105676","ENSG00000101361","ENSG00000144369","ENSG00000134802","ENSG00000243479","ENSG00000239880","ENSG00000180730","ENSG00000189007","ENSG00000173226","ENSG00000177191","ENSG00000117013","ENSG00000161179","ENSG00000134258","ENSG00000188883","ENSG00000130962","ENSG00000148843","ENSG00000132603","ENSG00000224785","ENSG00000080608","ENSG00000172175","ENSG00000114737","ENSG00000144228","ENSG00000104419","ENSG00000229644","ENSG00000116106","ENSG00000150990","ENSG00000076864","ENSG00000053524","ENSG00000125898","ENSG00000100304","ENSG00000136478","ENSG00000129667","ENSG00000123213","ENSG00000150672","ENSG00000172602","ENSG00000163938","ENSG00000135750","ENSG00000214706","ENSG00000109814","ENSG00000135333","ENSG00000155438","ENSG00000138764","ENSG00000159216","ENSG00000161638","ENSG00000144485","ENSG00000170927","ENSG00000120833","ENSG00000127995","ENSG00000185112","ENSG00000128059","ENSG00000151320","ENSG00000181744","ENSG00000108179","ENSG00000169184","ENSG00000158747","ENSG00000153714","ENSG00000154864","ENSG00000232044","ENSG00000117479","ENSG00000169884","ENSG00000278540","ENSG00000198093","ENSG00000198807","ENSG00000065485","ENSG00000166261","ENSG00000160208","ENSG00000083307","ENSG00000100644","ENSG00000140798","ENSG00000135763","ENSG00000227953","ENSG00000134575","ENSG00000215386","ENSG00000178896","ENSG00000122034","ENSG00000139318","ENSG00000004660","ENSG00000143367","ENSG00000162542","ENSG00000038427","ENSG00000215349","ENSG00000143333","ENSG00000107833","ENSG00000069509","ENSG00000113594","ENSG00000163704","ENSG00000197043","ENSG00000182580","ENSG00000159388","ENSG00000104936","ENSG00000278041","ENSG00000176402","ENSG00000142207","ENSG00000143158","ENSG00000140416","ENSG00000196917","ENSG00000169047","ENSG00000275993","ENSG00000074370","ENSG00000178573","ENSG00000101384","ENSG00000141441","ENSG00000090447","ENSG00000278921","ENSG00000166263","ENSG00000131188","ENSG00000168778","ENSG00000117394","ENSG00000154743","ENSG00000187210","ENSG00000141543","ENSG00000256043","ENSG00000136574","ENSG00000157193","ENSG00000151503","ENSG00000162545","ENSG00000130821","ENSG00000011028","ENSG00000184675","ENSG00000277161","ENSG00000140280","ENSG00000138442","ENSG00000162433","ENSG00000143324","ENSG00000241945","ENSG00000120594","ENSG00000188735","ENSG00000171208","ENSG00000129151","ENSG00000125895","ENSG00000182985","ENSG00000181444","ENSG00000181826","ENSG00000184564","ENSG00000111845","ENSG00000234912","ENSG00000174804","ENSG00000165905","ENSG00000133136","ENSG00000013588","ENSG00000033867","ENSG00000155893","ENSG00000164300","ENSG00000037897","ENSG00000145780","ENSG00000197358","ENSG00000069667","ENSG00000002834","ENSG00000169598","ENSG00000084676","ENSG00000156804","ENSG00000100100","ENSG00000139865","ENSG00000118894","ENSG00000143179","ENSG00000171631","ENSG00000149177","ENSG00000243701","ENSG00000261578","ENSG00000148459","ENSG00000120158","ENSG00000145912","ENSG00000275464","ENSG00000137575","ENSG00000174177","ENSG00000247271","ENSG00000156968","ENSG00000196950","ENSG00000281398","ENSG00000165175","ENSG00000166268","ENSG00000078401","ENSG00000138757","ENSG00000277363","ENSG00000187908","ENSG00000143797","ENSG00000126602","ENSG00000184304","ENSG00000228793","ENSG00000128829","ENSG00000110090","ENSG00000181458","ENSG00000248019","ENSG00000134247","ENSG00000005249","ENSG00000186193","ENSG00000196372","ENSG00000129116","ENSG00000198142","ENSG00000170421","ENSG00000092421","ENSG00000081189","ENSG00000232956","ENSG00000111846","ENSG00000157227","ENSG00000089127","ENSG00000162552","ENSG00000204060","ENSG00000164692","ENSG00000129625","ENSG00000166166","ENSG00000179958","ENSG00000070081","ENSG00000125454","ENSG00000049323","ENSG00000128512","ENSG00000137501","ENSG00000106799","ENSG00000167747","ENSG00000005884","ENSG00000160818","ENSG00000139793","ENSG00000105204","ENSG00000203805","ENSG00000105879","ENSG00000205413","ENSG00000138449","ENSG00000101444","ENSG00000246223","ENSG00000160193","ENSG00000166963","ENSG00000124788","ENSG00000160712","ENSG00000183779","ENSG00000168234","ENSG00000115053","ENSG00000257261","ENSG00000143061","ENSG00000232445","ENSG00000139946","ENSG00000102174","ENSG00000196391","ENSG00000126249","ENSG00000129514","ENSG00000189120","ENSG00000279233","ENSG00000177565","ENSG00000173559","ENSG00000240216","ENSG00000146083","ENSG00000080546","ENSG00000184916","ENSG00000222041","ENSG00000163359","ENSG00000264577","ENSG00000074582","ENSG00000137976","ENSG00000168743","ENSG00000164309","ENSG00000103264","ENSG00000187325","ENSG00000186187","ENSG00000177692","ENSG00000169926","ENSG00000113430","ENSG00000138778","ENSG00000102554","ENSG00000091436","ENSG00000162174","ENSG00000259345","ENSG00000143450","ENSG00000124787","ENSG00000173153","ENSG00000054118","ENSG00000133740","ENSG00000172594","ENSG00000196843","ENSG00000270179","ENSG00000245275","ENSG00000197208","ENSG00000134987","ENSG00000138018","ENSG00000105141","ENSG00000099812","ENSG00000225708","ENSG00000142178","ENSG00000113716","ENSG00000226745","ENSG00000141101","ENSG00000204428","ENSG00000137266","ENSG00000005893","ENSG00000138772","ENSG00000175928","ENSG00000090020","ENSG00000115468","ENSG00000165724","ENSG00000162139","ENSG00000136205","ENSG00000124493","ENSG00000134291","ENSG00000111961","ENSG00000175985","ENSG00000236088","ENSG00000102221","ENSG00000104228","ENSG00000250148","ENSG00000244468","ENSG00000121270","ENSG00000151576"],["SEC14L2","AQP3","C1orf116","RANBP3L","NFKBIA","CYP4F8","FKBP5","MYO1B","KMO","MYC","WWC3","DUSP10","CLDN8","UGT2B11","WNT7B","CNTNAP2","CHPT1","IRX3","TDO2","CARMIL1","KDM4B","UGT2B28","SNPH","PISD","SLC16A6","NADK2","TP63","CHST1","AKAP5","PARM1","XDH","TAF4B","HSD11B2","B3GNT5","SPRED1","PREP","MTCL1","CCDC86","GNA12","LARGE1","WWC1","AMIGO2","ADAMTS1","CROT","DDN","PDE4B","ADRA2A","AZGP1","NEDD4L","AR","PPP4R1L","NRARP","POLR3E","NOLC1","OTULINL","AZGP1P1","IQGAP2","DDX21","MYBBP1A","RALBP1","SP110","DOP1B","TMTC2","DOCK2","POU2F3","OSGIN2","UGT2B24P","TWNK","STS","TPRG1","DUSP2","PRR15L","PPFIBP2","FBXW7","FBN2","PRAG1","KLF15","ATP1B1","FAM83A","DKC1","ENC1","PEX10","WDR3","CXCR4","TOX3","MAFB","RAB27A","PTK2B","AGR2","PPAN","GADD45B","RRS1","C1orf115","NAGK","NOP16","GALNT10","SACS","SLC16A1","LINC02747","AFF3","AMPD2","LINC02233","WDR35","FHDC1","UMODL1","TMEM45B","UGT2B25P","TBC1D16","SLC35F2","TP53INP2","TP53INP1","BOP1","AZGP1P2","AL035661.1","ODC1","CYBRD1","DHCR24","AKAP1","GPC4","TNFSF10","CYP4F23P","PCAT14","SYBU","ARHGAP6","PER1","PHLDB2","NAV2","LYAR","NOL6","MYBPC1","RRP9","LYPD6B","INSIG2","ALDH1B1","LMCD1","RAB11FIP1","PUS1","SIX2","HABP2","OGFRP1","JUN","GLUL","EHF","POP1","SPATA13","PPARGC1B","GRB7","DPH2","NRK","MTHFD1L","PLCD4","TMEM164","CDKN1A","ID2","DGKH","PRKAA2","RABEP1","ZNF652","AC010735.2","BYSL","PRSS23","TENT5A","CITED2","SLC15A2","FSTL4","BMF","SLC16A2","SLC16A6P1","MYCN","RAB22A","ENTR1","PRKAB2","ARHGEF37","C4orf19","KALRN","EAF2","NXPE3","DST","RAC3","STC2","WDR43","ARHGAP18","TIPARP","PRPF19","GLI3","ALCAM","DDC","AFP","HS6ST2","TRIB2","RIOK1","CREB3L1","KRT23","TRIM14","INPP5J","PFKP","GADD45A","SRM","BLNK","ANKK1","RAB27B","TPST2","ERBB2","RBBP8NL","ENDOD1","AL162595.1","TSKU","FGD5","ACAA1","AC093904.4","RETREG1","TNS2","BAMBI","SIPA1L2","PRPS1","TAT","ABCC1","ACSL4","AC009119.1","POLR1B","MCCC2","RAP2C","MCC","CABLES1","FUT8","PIK3R3","UNC5B","FAM216A","TSR1","FAR2","TTC12","MACO1","FJX1","PLEKHS1","CRAMP1","KLF4","LINC00665","SLC22A5","RNF145","GBP2","CCDC88C","GEMIN5","SINHCAF","NAMPT","ACSL1","AL035425.3","DDB2","SLC16A14","LRCH1","DAPK2","PUS7","ADAMTS4","REEP1","ABCE1","RABL3","MBLAC2","EXOSC5","TNC","RNF152","TNFRSF19","IRX5","PPRC1","JPH1","PNO1","CDYL2","FLNB","PACSIN3","PEX5","SERHL2","MTFP1","SNHG17","SOX9","RAB4A","MARS2","FMOD","LYPD3","SH3PXD2B","AC007881.1","ATP2B4","ZFHX3","SCUBE3","TEX9","GALNT18","VASN","SPG21","CLMN","RND3","ZNF385A","CCDC85B","ERO1B","STK17A","AEN","KIF21A","CASP4","SRMS","LHFPL2","ABCA12","RALGPS2","BAG2","AC004816.1","SCYL3","EPHA2","WDR74","PFKFB2","AC004522.2","RCAN3","MARK1","ALKBH2","ARHGEF26","BCL6","IMP4","DNM2","ID2-AS1","SLC12A2","ERN1","MB21D2","AC007493.1","B4GALT5","NOC3L","ARMC6","NOP56","FAM171B","SLC43A3","MNX1-AS1","RALBP1P1","SHISA2","ADAT2","IQCB1","B3GNT8","KCNQ4","YDJC","VTCN1","KLRG2","PRRG1","PDCD11","NIP7","AC006026.1","PUM3","MALT1","CISH","SPOPL","NDRG1","NAMPTP1","EPHA4","DHX37","RAP1GAP","MCF2L2","FAM110A","TTLL12","TEX2","RHBDF2","NLN","DLG2","RND1","GNL3","KCNK1","IFRD2","UGDH","EPHA7","NIFK","CCNG2","RUNX1","ITGA5","HES6","PKHD1","SOCS2","CASD1","FAM43A","PPAT","AKAP6","DIPK2A","PPIF","MN1","NBL1","LURAP1L","PIEZO2","SILC1","SLC19A2","WNT10B","ACACA","ZNF649","PAX9","PDIA5","ZNF202","RRP1B","GRHL2","HIF1A","ABCC12","URB2","LINC01341","ACP2","MIR99AHG","EXOSC4","GTF3A","DUSP6","CAMKK1","TUFT1","TMCO4","VCAN","MRPL3P1","RGS16","NPM3","FUNDC1","LIFR","PRRT3","ANXA6","EPHB3","BTG2","DMPK","AL133325.3","GJC3","URB1","MPC2","TPM1","HCAR1","IRS1","SIK1B","ATP2A3","MAF","JAG1","GAREM1","TFAP4","EPB41L4A-DT","STXBP4","PRR7","TCTN2","SLC2A1","TSEN2","GCNT1","EIF4A3","CTSO","GATA4","LRP8","NCAPD3","CAMK2N1","SLC6A8","MRC2","AMER1","PIGW","LYSMD2","WDR12","AK4","XPR1","PWP2","PLXDC2","TMEM120B","NETO2","BBOX1","TMEM74B","CADM1","ZNF467","RELL1","SLITRK6","PAK1IP1","SNHG20","FZD4","LARGE2","GNG5P2","GPRC5A","SLC4A7","PXYLP1","SERINC5","METTL1","FEM1C","BNIP3P1","RORA","LASP1","DFFB","NCOA1","FBXO32","PIK3IP1","TTC6","EEF2KMT","UCK2","P2RY6","PTPRJ","DUBR","AP003119.3","PDSS1","RCL1","NHP2","FP565260.1","SDCBP","CTU2","ZBED5-AS1","MPV17L","SLC39A10","SNHG4","MID1IP1","MYRFL","EDN1","G3BP2","SRCIN1","DMBT1","MBOAT2","TRAP1","PRKD1","AL033523.1","EIF2AK4","CPT1A","TMEM45A","FAM13A-AS1","PTGFRN","PRKAR2B","SAPCD2","ASB13","PALLD","SOWAHC","KRT8","SEMA6A","MEF2C","SNHG15","GCNT2","MMP14","OAS1","WNT4","FOXO6","COL1A2","REEP5","TRMT61A","DCTPP1","NUCB2","SLC25A19","LTBP1","DOCK4","SYTL2","TGFBR1","C19orf48","ITGA3","GPATCH4","MBNL2","DYRK1B","PLPP4","CBLL1","SAMD9","SLC40A1","AHCY","LINC01550","WDR4","MAP1A","ATXN1","IL6R","ZNF703","TTC39C","NCL","AC008014.1","IGSF3","EMSLR","PELI2","PHEX","ZNF774","PDCD2L","FOXA1","SP6","AC122688.3","TBL1XR1","NABP1","CPHL1P","RNF44","SESN1","JAG2","CYTOR","COL6A3","AC010761.1","BCS1L","DNASE2B","NPNT","CMYA5","FBXO31","TAF9B","ZNRF1","DNAJC28","KLF13","IRX4","CENPE","KLF5","MAP3K20","ASRGL1","AC013652.1","OAZ3","RPP40","ESRRA","THRAP3","E2F5","SMPDL3A","ARID5A","AP002840.2","SAP30L-AS1","SLC22A4","WDR36","SELENOI","CASP14","MISP","AL354984.2","SIK1","HMGXB3","OAZ1P1","NOB1","LY6G5C","SLC22A23","LAMP2","ANXA3","LRRN1","SLC9A1","EFHD1","ZMYND19","NEU3","TNS3","GRM4","TMEM106C","SASH1","PLEKHD1","COX10-AS1","JADE3","TRIM35","KRT8P31","AC093001.1","ABCC11","QTRT2"],[7.47,7.68,4.16,4.51,7.68,4.62,7.76,4.99,5.7,6.48,8.41,5.26,4.88,3.25,5.93,6.75,6.36,7.62,6.8,6.09,6.84,2.26,7.16,6.16,8.18,6.61,2.37,5.8,5.82,5.15,4.59,2.26,4.21,5.16,7.23,6.69,4.19,5.67,7.92,6.34,6.96,6.23,5.56,5.08,2.18,5.26,1.93,9.6,6.7,8.550000000000001,2.73,6.23,6.8,8.1,2.97,7.07,5.65,7.82,5.37,6.59,5.03,6.82,5.36,3.12,5.16,7.5,0.6,4.76,7.53,2.82,3.92,7.21,6.5,7.02,2.48,5.27,2.15,6.55,2.4,8.98,4.75,4.9,6.26,4.05,5.85,6.95,5.33,4.11,8.66,3.61,4.3,6.1,6.81,6.99,4.84,8.289999999999999,4.01,3.84,3.04,2.78,5.5,2.94,4.24,4.67,2.82,7.63,0.68,8.130000000000001,4.81,4.79,6.53,7.19,1.25,3.08,6.7,6.41,10.21,5.75,4.01,3.91,2.48,1.5,3.58,2.86,4.17,2.6,2.41,4.67,6.85,1.41,5.2,3.93,7.02,4.68,4.99,5.45,5.15,8.210000000000001,1.7,3.88,5.13,6.99,9.31,5.7,5.49,4.01,6.04,5.6,6.17,4.44,4.01,6.38,4.61,8.029999999999999,6.9,5.07,7.09,7.14,2.56,5.28,6.11,5,7.09,3.13,2.49,3.6,6.22,3.91,1.65,7.2,6.37,4.92,5.25,5.35,2.92,3.97,5.37,5.52,6.29,4.29,6.1,4.71,5.41,8.23,6.22,8.16,2,0.73,5.95,5.87,5.12,3.61,1.56,5.49,6.05,6.79,4.45,7.11,3.96,2.7,7.33,5.62,9.960000000000001,3.71,3.23,6.46,8.390000000000001,3.27,5.9,1.88,3.12,4.39,4.97,7.32,5.81,2.38,5.42,1.97,1.26,6.08,8.1,7.36,5.69,5.54,7.24,7.98,6.14,3.48,5.43,3.03,5.96,6.3,5.24,2.59,5.15,4.48,5.04,5.54,3.12,2.85,4.77,5.63,5.62,4.7,8.9,4.43,4.52,6.28,6.09,3.12,4.53,3.34,4.51,7.18,6.45,4.05,4.98,8.9,5.6,1.33,7.38,6.15,4.93,5.1,5.18,7.36,5.53,5.87,1.92,3.53,5.58,3.46,7.11,4.54,6.39,5.66,2.62,2.87,6.34,6.28,1.55,7.33,3.61,5.09,7.77,5.71,5.78,6.84,5.75,4.18,4.88,5.68,6,3.91,4.69,2.72,7.37,6.39,2.64,4.13,5.52,2.96,5.19,5.49,1.15,4.54,1.3,4.91,5.75,4.34,6.65,8,3.17,6.72,5.62,4.95,3.02,5.73,5.15,6.24,8.09,1.19,6.17,3.26,1.99,2.08,2.93,4.91,2.44,1.55,6.08,2.6,4.69,3.53,6.4,5.05,2.2,4.82,5.43,5.51,7.19,7.51,3.66,5.57,5.77,7.59,2.55,6.89,8.529999999999999,8.4,4.71,5.54,0.75,5.11,7.65,5.15,6.52,6.03,1.79,5.55,7.67,6.71,0.97,6.19,1.06,6.08,7.18,4.15,5.27,2.26,4.14,7.22,0.7,7.68,2.65,0.92,5.37,5.59,2.19,7.32,4.61,4.68,5.34,4.34,5.9,7.28,6.83,3.01,5.96,2.5,5.66,1.84,6.06,7.77,7.39,2.14,6.24,4.73,1.38,3.64,1.56,5.03,5.25,1.83,4.34,7.87,6.14,6.91,5.13,2.16,2.89,5.78,7.08,6.37,5.6,6.5,1.21,3.31,1.89,5.31,4.45,4.44,2.44,5.4,4.26,2.66,6.57,3.72,2.38,8.050000000000001,3.73,2.98,6.57,7.77,7.1,6.63,4.13,3.59,4.87,3.46,5.95,6.97,6.72,5.08,1.74,4.15,7.44,1.98,1.9,4.02,5.68,3.4,4.01,5.22,3.15,6.29,2.05,3.33,5.83,5.22,4.17,4.75,4.45,6.22,5.77,4.11,8.19,4.07,5.52,5.62,3.24,4.63,3.8,6.88,5.08,5.83,3.33,4,3.8,4.31,5.36,4.94,6.33,4.06,2.52,5.66,4.9,2.46,6.06,2.33,3.52,8.43,5.08,3.51,7.42,6.87,1.85,3.71,7.13,4.8,1.36,2.61,5.86,2.3,8.279999999999999,3.75,3.82,4.83,8.77,0.96,2.18,5.46,3.69,1.05,3.5,2,3.88,1.45,9.19,5.21,7.51,5.63,6.07,4.79,4.51,4.55,4.04,7.28,7.05,6.6,3.62,5.26,5.86,4.29,2.47,8.720000000000001,8.92,1.46,4.32,1.08,5.98,3.82,5.81,3.08,9.76,0.74,6.67,1.43,1.88,0.95,1.81,3.46,8.93,6.84,1.86,8.220000000000001,4.48,2.57,4.96,4.93,7.08,4.48,1.93,1.71,5.45,1.22,7.38,4.22,4.91,5.89,5.2,1.23,4.22,3.94,6.18,2.6,6.75,2.13,1.13,3.53,4.25,5.38,7.89,4.13,5.81,3.83,1.94,3.06,2.59,6.18,7.05,3.3,1.08,2.46,1.05,6.63,4.9,4.7,2.17,5.7,7.61,1.49,5.08,6.32,7.12,7.19,5.54,4.67,5.71,8.130000000000001,7.75,2.63,3.21,6.34,3.63,1.47,5.73,7.91,5.05],[5.37,4.07,3.24,4.21,1.85,5.32,2.05,2.38,-1.56,1.84,1.57,1.69,3.13,4.4,2.18,1.15,1.56,1.14,-1.63,1.36,1.08,5,-1.32,1.07,1.12,1.17,-2.59,-1.14,1.06,1.1,-1.57,1.66,1.32,-1.19,0.86,1.23,1.19,1.26,0.87,1.05,0.99,-1.15,1.45,1.54,1.73,-1.1,1.95,1,0.88,-0.91,-2.65,0.98,0.91,0.82,1.56,1.19,1.46,0.82,0.9,0.8,-0.89,0.97,0.9399999999999999,-2.1,-0.96,-0.84,5.72,0.83,-0.88,-1.89,1.18,1.03,-1,-0.86,-1.56,1.11,1.65,1.02,-1.52,0.88,-0.89,0.89,0.8,-1.54,1.05,0.99,0.98,1.05,-0.97,0.92,0.84,0.8,-0.97,0.83,1.07,-0.93,0.91,1.16,-1.44,1.49,0.73,-2.34,0.85,0.87,-1.51,0.71,3.74,0.85,0.8100000000000001,-0.82,-0.73,0.65,1.49,-0.9399999999999999,0.64,-0.73,0.73,0.8,-0.88,-0.89,1.58,1.67,1.24,1.12,0.88,-1.22,1.19,0.77,0.6899999999999999,3.51,0.85,-0.93,-0.6899999999999999,0.76,-0.68,1.07,0.74,-0.74,-1.97,0.83,0.84,0.6899999999999999,0.88,0.68,0.77,1.11,-0.64,0.75,0.67,0.78,-0.85,0.74,-0.8,-0.83,0.62,0.73,-0.6,0.83,-1.1,0.7,0.62,0.65,0.76,1.05,-1.01,-1.15,-0.64,1.19,-1.54,-0.72,0.6899999999999999,0.79,0.68,-0.77,-0.9,0.97,0.7,-0.8,0.61,1.15,0.76,0.8100000000000001,0.74,0.65,-0.9,0.8100000000000001,1.45,-2.5,0.62,-0.6899999999999999,0.64,-1.08,-1.93,0.7,-0.62,0.59,0.79,0.63,-0.71,1.16,-0.61,-0.64,-0.6,-0.8,0.88,0.6899999999999999,0.54,-0.91,0.72,-1.79,0.85,-0.6899999999999999,-0.77,-0.79,0.65,1.21,0.62,1.08,-1.2,0.61,0.7,-0.63,-0.79,-0.62,-0.6,-0.61,0.75,0.88,0.59,1.09,0.59,-0.6,0.76,-0.87,0.64,0.84,0.62,0.5600000000000001,0.8,-1.24,0.73,0.62,0.68,0.64,0.63,-0.61,-0.75,-0.65,0.61,-0.77,0.63,-0.75,-0.6,0.53,-0.61,0.79,0.66,-0.8,-0.78,1,0.54,0.53,0.72,0.58,-0.66,0.62,-0.62,0.57,0.87,0.93,0.57,-0.82,0.5600000000000001,0.74,-0.73,-0.64,0.83,0.89,-0.54,0.59,-1.3,-0.58,0.74,-0.59,0.52,0.67,0.78,-0.63,0.6,0.65,-0.59,0.65,0.61,-0.61,0.83,-0.75,-0.7,-0.54,1.09,-0.63,-0.54,0.87,0.5600000000000001,0.63,1.22,-0.6,-1.16,0.66,0.52,0.91,0.58,0.59,-0.6899999999999999,-0.5600000000000001,0.84,-0.63,-0.73,0.58,0.64,0.52,0.49,-1.13,0.5600000000000001,0.85,0.9399999999999999,-1.22,0.77,0.5600000000000001,-0.86,1.06,0.58,-0.82,0.66,0.63,0.53,0.55,0.75,0.59,0.5600000000000001,-0.54,-0.58,1.21,0.63,-1.25,0.52,0.5,-0.83,-0.53,0.58,0.55,0.63,0.52,-1.17,-0.53,0.54,0.62,0.59,0.8,0.96,0.57,-0.58,0.62,-1.2,0.6,1.14,0.5600000000000001,0.65,-0.79,0.5600000000000001,-0.78,0.61,0.49,-1.44,0.67,-0.76,-1.38,-0.58,0.58,0.84,0.57,0.55,-0.62,0.52,0.61,0.48,0.53,0.5600000000000001,1,0.6,-0.95,-0.61,-0.98,0.59,0.57,-0.6,0.99,0.51,-0.55,-1.62,0.61,1,0.55,-0.65,1.16,-0.67,-0.48,-0.5,-0.53,-0.52,0.93,0.95,0.54,0.95,0.51,-0.49,-0.58,0.87,-0.64,-1.17,0.72,-0.51,0.72,-0.97,0.61,0.57,-0.6899999999999999,0.48,0.61,0.88,0.48,-0.76,0.66,0.5600000000000001,0.6,-0.5600000000000001,0.52,0.57,0.67,0.57,0.7,0.54,0.49,-0.55,0.52,-0.85,0.52,0.49,-1.06,-0.95,0.55,-0.5,-0.8,-0.82,0.62,0.68,0.48,0.9,0.72,0.59,0.62,0.55,0.58,0.61,-0.48,0.63,0.58,-0.46,0.62,0.5600000000000001,0.55,-0.59,-0.67,0.5600000000000001,0.46,-0.54,0.49,0.62,0.63,0.64,0.62,0.48,0.51,-0.5,0.53,-0.8,1.01,0.57,0.93,0.5,-0.88,-0.9,0.63,-0.59,-0.73,0.48,0.64,-0.8,0.55,0.53,-0.53,-1,0.72,-0.49,0.82,0.5,0.61,-0.58,0.57,-0.43,-1.03,-0.7,0.8,-0.55,-1.01,-0.6,0.89,-0.57,-1.16,-0.48,0.6899999999999999,0.49,-0.54,0.49,-0.53,0.7,-0.85,-0.5600000000000001,0.44,-0.47,0.45,-0.58,-0.53,0.49,0.61,-0.64,-0.78,0.54,-0.9399999999999999,0.51,-1,-0.58,0.62,0.6,0.55,0.47,-1.13,-0.44,0.93,-0.98,-0.96,-0.92,0.5600000000000001,-0.71,0.5600000000000001,0.73,-0.46,0.57,-0.93,0.65,-0.52,0.43,-0.65,-1.41,0.78,0.46,1.36,-0.51,-0.59,0.51,-0.62,0.59,-0.84,0.52,0.5,-0.5600000000000001,0.8,-0.46,0.73,-0.98,-0.55,0.52,0.53,0.42,0.49,0.46,-0.55,0.73,-0.85,0.57,0.5,0.42,-0.83,-1.14,-0.85,0.76,0.44,-0.78,0.5600000000000001,-0.7,0.47,-0.42,-0.92,-0.6,-0.5,0.44,0.47,0.46,0.5600000000000001,-0.46,-0.43,0.59,0.6899999999999999,0.6899999999999999,-0.49,0.49,-0.79,-0.53,0.54,0.59],[1.190620574159139e-13,4.505787006827785e-12,6.42712021507107e-12,1.245405617131997e-11,1.716132224314883e-11,2.47595353828963e-11,6.648746559640377e-11,2.042536963672045e-10,2.175956357965751e-10,2.179507052207437e-10,3.895133755782472e-10,3.983027775903215e-10,7.483223772481638e-10,9.553260908794977e-10,1.400956529109084e-09,4.274264774044282e-09,5.609114754548351e-09,8.273315954591335e-09,8.450929689270428e-09,1.019910063979706e-08,1.045369770265331e-08,1.127205933460101e-08,1.493312874300781e-08,1.756498869492461e-08,2.249119392597575e-08,2.336167438918609e-08,2.688593884833731e-08,2.944520358549797e-08,2.965651592965299e-08,3.365060551865713e-08,3.423336003972747e-08,3.470692744729494e-08,3.478664161342634e-08,4.340088012818795e-08,5.262828503529803e-08,5.840966082644907e-08,6.261208199480921e-08,6.588739146590629e-08,6.623619182296189e-08,6.660148452063098e-08,7.498488552546828e-08,8.717040475636027e-08,9.075185876639531e-08,9.464259896833448e-08,1.001483714852636e-07,1.045652784903635e-07,1.19797723009529e-07,1.609167480393487e-07,1.65554046901337e-07,1.665052024791475e-07,1.908631968082029e-07,1.963088435246195e-07,1.974524589373515e-07,2.146199797204976e-07,2.212253908117014e-07,2.478341403205767e-07,2.765595413294731e-07,2.792118426344082e-07,2.987377243323841e-07,3.039750452248971e-07,3.284285927902401e-07,3.843019651261583e-07,4.036826201818897e-07,4.087508862194718e-07,4.28614584849737e-07,4.372814492822397e-07,4.982041751231244e-07,5.001549912113545e-07,5.052151656882436e-07,5.504617819462936e-07,5.627678141132209e-07,6.493146199834535e-07,7.114018866263349e-07,7.379535114426096e-07,7.752973495999934e-07,7.821755390880418e-07,7.954326041400568e-07,8.449334567691692e-07,8.505489024418308e-07,8.554478114596616e-07,9.396244943096595e-07,9.573751911456584e-07,9.738476936298008e-07,1.058503961621739e-06,1.087353275472989e-06,1.098573139542967e-06,1.11100358069445e-06,1.190031236755647e-06,1.214328761173652e-06,1.272707757256576e-06,1.333508415171685e-06,1.428569471616153e-06,1.495360581055317e-06,1.533028007845888e-06,1.565370502147989e-06,1.571460415122597e-06,1.576785791923779e-06,1.586632646477149e-06,1.628173805774886e-06,1.689256959754728e-06,1.792272954867476e-06,2.089348650918387e-06,2.096049074686851e-06,2.161693376491246e-06,2.212518351701459e-06,2.281404487787715e-06,2.398291823660631e-06,2.475576949759765e-06,2.499396176000178e-06,2.554111591468618e-06,2.614657067375745e-06,2.715919934870089e-06,2.742962809200795e-06,2.761837604381458e-06,2.822327378624874e-06,2.916817475699565e-06,3.021818023827356e-06,3.093695242509393e-06,3.441567354974923e-06,3.584251554081592e-06,3.63049433321812e-06,3.697238885982204e-06,3.710468666737855e-06,3.898640314183747e-06,3.911312336580528e-06,4.063695295421041e-06,4.066637783408977e-06,4.240053143443106e-06,4.360328719172187e-06,4.712485294363802e-06,5.00227690443792e-06,5.166682487949535e-06,5.210185227921161e-06,5.253458796382017e-06,5.533095203144438e-06,5.60108276729827e-06,5.665993680952173e-06,6.066675164010522e-06,6.096120865599431e-06,6.388435912136206e-06,6.423011283579718e-06,6.800518759077617e-06,6.814109659759879e-06,6.894575760585784e-06,6.947296932899822e-06,7.013858048355124e-06,7.644539288837835e-06,8.339263725072511e-06,8.491423565476889e-06,8.832045248374932e-06,8.951570691841121e-06,9.365474320257051e-06,9.40997475037364e-06,9.428809274001155e-06,9.695908375498232e-06,9.900930947949922e-06,1.001613147505779e-05,1.006622749228436e-05,1.02497265846906e-05,1.037583501200698e-05,1.050488716528191e-05,1.052120789279042e-05,1.052215602031515e-05,1.075428505527933e-05,1.102718235293253e-05,1.128259768529904e-05,1.137999550812835e-05,1.162219516438839e-05,1.191747573097269e-05,1.195573991023551e-05,1.201386525983394e-05,1.328223305533057e-05,1.405827710846165e-05,1.412942049647694e-05,1.519427546978529e-05,1.602148375927116e-05,1.613213187308424e-05,1.683386615702778e-05,1.689010724898233e-05,1.704963469075863e-05,1.726737298050026e-05,1.849504955989624e-05,1.874928133275841e-05,1.881545722382071e-05,1.894891546039596e-05,1.938659787488761e-05,1.971914544268163e-05,2.043384474206636e-05,2.096967287226414e-05,2.113827908719417e-05,2.141977108198984e-05,2.228153032749723e-05,2.229300503614871e-05,2.253376196096207e-05,2.268476266639605e-05,2.311914898415119e-05,2.384780757262414e-05,2.497193154007766e-05,2.527368314620562e-05,2.542024315991779e-05,2.61934109397887e-05,2.6508024429752e-05,2.661165343626405e-05,2.736455547547553e-05,2.740826201302791e-05,2.771531938212046e-05,2.777119048583222e-05,2.812692481384303e-05,2.853092761101117e-05,2.864874568715702e-05,2.882844145529428e-05,2.961899165690367e-05,3.011093918339713e-05,3.020962125717886e-05,3.359809631591341e-05,3.364734098406293e-05,3.48289097985311e-05,3.495737947540003e-05,3.525824769227073e-05,3.650188518247916e-05,3.692290841092663e-05,3.703420393502543e-05,3.737143361467661e-05,3.818167814546777e-05,3.866321817188806e-05,3.884743887173639e-05,3.959818721288483e-05,3.994362510512268e-05,4.192618497166021e-05,4.228400359790522e-05,4.239149876540536e-05,4.318289602378573e-05,4.324846374344678e-05,4.373347537133074e-05,4.428766039654243e-05,4.441970766239822e-05,4.531358382941705e-05,4.595922530527205e-05,4.659134784106091e-05,4.76042968533297e-05,4.776646583858167e-05,4.808366130602024e-05,5.196338209227251e-05,5.247962831003821e-05,5.254566298907882e-05,5.394112824904665e-05,5.446476172975039e-05,5.58366537062048e-05,5.905950105145528e-05,5.956924157578523e-05,5.95798736137167e-05,6.076810746930507e-05,6.131688998649176e-05,6.192092774998621e-05,6.313611334127893e-05,6.342456669871158e-05,6.359530215124798e-05,6.470929100521727e-05,6.632014210669504e-05,6.732205196325384e-05,6.732312442596648e-05,6.733591544117856e-05,6.786128810887211e-05,6.879166815559342e-05,6.911277545510533e-05,6.92643041520868e-05,7.017674974281866e-05,7.233976103258546e-05,7.368887615528655e-05,7.380657635234423e-05,7.854842477125106e-05,7.857061457574418e-05,7.989258830302064e-05,8.054518015819832e-05,8.092757651083186e-05,8.171047839043636e-05,8.261218410190245e-05,8.31002649376321e-05,8.324250190353546e-05,8.362560512180335e-05,8.434702169434451e-05,8.496372191597002e-05,8.535000715111823e-05,8.803770237862169e-05,8.933964048819721e-05,9.034506242933003e-05,9.350498755304892e-05,9.397653182850278e-05,9.62177210539229e-05,9.628054656019072e-05,9.904490871633773e-05,0.0001004262262888354,0.0001007026232425428,0.0001012492947700776,0.0001013002931640559,0.0001013452956995198,0.0001034840229488742,0.0001047047463510466,0.0001052126330326727,0.0001060291617650952,0.0001071274949781611,0.0001076250534148519,0.0001082918832523703,0.0001088698353306529,0.0001092023805220114,0.0001109208661023582,0.0001133158593182582,0.0001137289845079598,0.0001150879394777235,0.0001152579872829619,0.0001152641943243804,0.0001170328687724994,0.0001197629858780692,0.0001201482608521144,0.0001218545250305581,0.0001233030534718748,0.0001237033385969341,0.000124189767184617,0.0001242431139481248,0.0001255619610577711,0.0001256992659905566,0.0001273033465554264,0.000129568586124981,0.0001329385904589573,0.0001335306816822654,0.0001349216692512304,0.0001364725917156912,0.0001375573161539088,0.0001388302875116286,0.0001402099136988059,0.0001418776668633352,0.0001420735186517886,0.0001422136373848212,0.0001427886709361641,0.000144498854859869,0.0001447333017708321,0.0001456719190342466,0.0001490902087058307,0.0001493930867513112,0.000149502669442065,0.0001506540545013988,0.0001531902835884254,0.0001534655142380751,0.0001586496387401741,0.0001595826313410608,0.0001598917443604567,0.0001600468405403351,0.0001621018657679849,0.0001674447480049241,0.0001691668873455354,0.0001780046204138046,0.0001784514124249205,0.000180803237658559,0.0001829796901747378,0.0001839848657689873,0.0001872006404693449,0.000188625729054049,0.0001918981327757589,0.0001937313563968101,0.0001938135377574206,0.0001952497558085082,0.0001966872690650516,0.0001966950990613825,0.0001974461857099996,0.0001997640215841267,0.0002002646024540839,0.0002005351204096438,0.0002011840436362233,0.0002079217399408071,0.0002079451709569748,0.0002118368155121511,0.0002121821551571491,0.0002165347476690892,0.0002173654267148318,0.0002174222112519252,0.0002195460615964392,0.0002199727895018983,0.0002220508028713256,0.0002245299550279806,0.0002249697061544413,0.0002252251001075062,0.0002282338667754671,0.0002288276762462457,0.0002304711426616218,0.0002322217049833146,0.000233076913256992,0.0002343164364451609,0.0002370450777671985,0.0002388020061605199,0.0002440724525974806,0.0002474873517122822,0.0002488160922660332,0.0002498889862784089,0.0002525983651351722,0.0002535625301529993,0.000254208731317935,0.0002554705364558264,0.0002584863712350401,0.0002663921135370363,0.0002669065491585834,0.0002764639535258153,0.0002773436490638111,0.0002782400616941913,0.0002795877097930326,0.0002824518067344826,0.0002864915456998606,0.0002878759125889418,0.0002980043255049434,0.0002989517798592891,0.0003043311343196414,0.0003044481370190924,0.0003046905749792194,0.0003073382531941954,0.000307573558361423,0.000307956891575656,0.0003100726578496683,0.0003109891206970769,0.0003143599784170184,0.0003155201185617582,0.0003157298628743769,0.0003171794388877816,0.0003290435083243093,0.0003359229394175518,0.0003365252680958267,0.0003420601479686917,0.0003427928099999786,0.0003445853165304402,0.0003448506308669372,0.0003521547551174115,0.0003523660706597519,0.000355664059450461,0.0003582243421369768,0.0003719450160766305,0.0003740179105180598,0.0003773128863608664,0.0003865302417456398,0.0003965529269935567,0.0003968976003372603,0.0003982138660325289,0.0003985459520027383,0.0004062228122161238,0.0004071530569282806,0.0004080387642466035,0.0004161198388358458,0.000427861204935211,0.0004286044542732911,0.0004311491234449763,0.0004333580221236486,0.0004382089282321919,0.0004417282298003129,0.0004473669471149087,0.0004485154819959833,0.0004486240733627902,0.0004487973288186041,0.0004559775939356554,0.0004562828715856437,0.0004600186402643303,0.0004616925090282332,0.0004650349489473914,0.0004726397287977436,0.0004747399110079914,0.0004802419171369119,0.0004864556189208063,0.0004901054308697253,0.0004946384798753221,0.0004956028614885715,0.0004962956653832754,0.0004998295034986796,0.0005010233149849793,0.0005310496026782844,0.0005357206710267807,0.000542775571408239,0.0005528528329449417,0.0005552720943890136,0.000563085345906795,0.0005701693911122641,0.0005882736358231563,0.0005918305746331098,0.0005924711703660288,0.0005977353097527249,0.0006010206762283815,0.0006035418330021131,0.0006036177211809022,0.0006036404933594993,0.0006091715247820531,0.0006285972293734624,0.000634902489639339,0.0006621674560927838,0.0006808382402758469,0.000684030299043638,0.0006959760683832834,0.0006977273641793723,0.0007029164675171169,0.0007133578085059605,0.0007147537498906998,0.0007220554247675598,0.0007249695277944531,0.0007286952261127445,0.0007392224966558392,0.0007559292581406212,0.0007605080882302315,0.0007627294980693077,0.0007804506961887131,0.0007893020306623369,0.0008057172681147829,0.000806622685972245,0.0008155514636696645,0.0008224274693109052,0.0008251899190039194,0.0008332702774514611,0.0008402371461017811,0.0008534725552633829,0.0008750551762431139,0.0008820108379941056,0.0008879119216107621,0.0008994396362828598,0.0009182569392531494,0.0009205997816273956,0.000926746381060324,0.0009276540352778275,0.0009291874218635367,0.0009477148139274721,0.000957672729948468,0.0009801274560739203,0.0009832534998958601,0.0009924555836151932,0.00100121428366711,0.001004053080751311,0.001014656486880997,0.001016192534400172,0.001020574687185006,0.00104227605950757,0.001055880522481476,0.001056030590657012,0.001083534724515605,0.001087391121887072,0.001105728465481619,0.001118919671039454,0.001130894652965107,0.001130954072076783,0.001142893458731051,0.001150242890866141,0.001153981106151384,0.001156076086313634,0.001159548866314541,0.001169597349495785,0.001170877578557481,0.0011728760637994,0.001178349234756978,0.001192782220082614,0.001203980563743345,0.001204809765317178,0.001206391416199617,0.001221863842771769,0.001255470076091351,0.001257263686385054,0.001271233913990301,0.001277318795295678,0.001305922283237646,0.001306053212556788,0.00130971887119713,0.001333542476489336,0.001354598153808883,0.001368030963238738,0.001391431547736374,0.001433287218268533,0.001451789183250712,0.001451974091318765,0.001462153512311784,0.001462783793332742,0.001506136643517337,0.001528566904078318,0.001548380826643372,0.00154845480231058,0.001554037589264261,0.001561375658651447,0.001562024685893549,0.001568196336538508,0.00159735927731553,0.001601912637881285,0.001608933448431012,0.001620031637852165,0.001628480813081015,0.001668687527542139,0.001688390630985695,0.001690859668569426,0.00171068806558323,0.001727548853730065,0.00173139359976596,0.001742616625270254,0.001746759305364422,0.001754248512687618,0.001760615840494723,0.00176699603725736,0.001786888309825448,0.001817957360824278,0.001837185664365746,0.001847014786192244,0.001848011414617695,0.001849195737566018,0.001878231530962699,0.001889239843793127,0.001893135493282442,0.001915020124494825,0.001928999436019214,0.001931831685790591,0.001932776567987055,0.001939967454274092,0.001966986502664105,0.001991425087563511,0.002004759757644241,0.002027536922826139,0.002038246473653109,0.002050699457824528,0.0020555393712733,0.002073291344878619,0.00209594649391181,0.002114642117492029,0.002116754833229059],[1.73342449391829e-09,3.11908144037399e-08,3.11908144037399e-08,4.532965094956187e-08,4.997033810760076e-08,6.007901260659786e-08,1.382844302311489e-07,3.173144317308808e-07,3.173144317308808e-07,3.173144317308808e-07,4.832408449114576e-07,4.832408449114576e-07,8.380634992581551e-07,9.934708969367576e-07,1.359768407153277e-06,3.889313802831919e-06,4.803711865380555e-06,6.475636070846746e-06,6.475636070846746e-06,7.247399278710934e-06,7.247399278710934e-06,7.459541447838916e-06,9.452670494323945e-06,1.065536126705864e-05,1.308163913200617e-05,1.308163913200617e-05,1.449749569233122e-05,1.488859363516613e-05,1.488859363516613e-05,1.534723379545073e-05,1.534723379545073e-05,1.534723379545073e-05,1.534723379545073e-05,1.858451217018495e-05,2.189186290939726e-05,2.362184033256311e-05,2.424127532839666e-05,2.424127532839666e-05,2.424127532839666e-05,2.424127532839666e-05,2.662694996012909e-05,3.021699816304403e-05,3.072689097162673e-05,3.131594541772685e-05,3.240133645453229e-05,3.309491064220003e-05,3.710925636799432e-05,4.848298485787816e-05,4.848298485787816e-05,4.848298485787816e-05,5.423981791828115e-05,5.423981791828115e-05,5.423981791828115e-05,5.78639311990875e-05,5.856037208777384e-05,6.443245087370135e-05,7.008698649852327e-05,7.008698649852327e-05,7.371733099246068e-05,7.375954472382129e-05,7.838675217103453e-05,9.024277919793127e-05,9.298443988233266e-05,9.298443988233266e-05,9.600307293580493e-05,9.646031242575951e-05,0.0001066003999602194,0.0001066003999602194,0.0001066003999602194,0.000114488186905087,0.0001153991071221744,0.0001312968271158208,0.0001418808228409974,0.0001451873672039589,0.0001498380746524053,0.0001498380746524053,0.0001503987439438323,0.0001556808085880152,0.0001556808085880152,0.0001556808085880152,0.0001688888026253621,0.0001699807976571908,0.0001708222719464611,0.0001834614187767964,0.0001859207026589713,0.0001859207026589713,0.0001859207026589713,0.0001968825542718803,0.0001986450835272719,0.0002058816915322055,0.0002133466924888413,0.0002260711188832562,0.000234096287092305,0.0002357120887761307,0.0002357120887761307,0.0002357120887761307,0.0002357120887761307,0.0002357120887761307,0.0002394402266492582,0.0002459389207706908,0.0002583534846526296,0.0002962755192074356,0.0002962755192074356,0.0003026162871955389,0.0003067814731659195,0.0003133487541292579,0.0003263245856137862,0.0003337215260328927,0.0003338413662971247,0.0003380482787290147,0.0003429440742695809,0.0003527157340542952,0.0003527157340542952,0.0003527157340542952,0.000357306646133909,0.0003660857381785342,0.0003760226376829271,0.0003817043138618157,0.0004210569674040328,0.0004348593197989492,0.0004368294793167158,0.0004391927912116782,0.0004391927912116782,0.0004555583704662072,0.0004555583704662072,0.0004661903896744196,0.0004661903896744196,0.0004822729196514702,0.0004921087273056424,0.000527762103081866,0.0005559400721504708,0.0005698615935004339,0.0005703389979947682,0.000570784377735267,0.0005967135782413324,0.0005996041471257024,0.0006021255620509685,0.0006385138394407346,0.0006385138394407346,0.0006632100799832419,0.0006632100799832419,0.0006937526051499586,0.0006937526051499586,0.0006970703367942252,0.0006975565244557827,0.0006994161597671387,0.0007571214116067349,0.0008203468957657479,0.0008297089643609264,0.0008572383118072708,0.0008630855476987741,0.0008913898325985897,0.0008913898325985897,0.0008913898325985897,0.0009107272905734113,0.0009240234209692495,0.0009275582662035945,0.0009275582662035945,0.0009385268512359148,0.0009398286472378417,0.0009398286472378417,0.0009398286472378417,0.0009398286472378417,0.0009547050982915354,0.0009729984719778467,0.0009895381909654746,0.000992103919777489,0.001007187734513872,0.001022864703613581,0.001022864703613581,0.001022864703613581,0.001124279250305569,0.001182242718437975,0.001182242718437975,0.00126407689465488,0.001325322625291073,0.001326936203052167,0.001373760175630915,0.001373760175630915,0.001379031285904194,0.001388926426646979,0.001479502343640272,0.001488773052834813,0.001488773052834813,0.001491228433448134,0.001517470314303703,0.001535246195187176,0.001582427370211404,0.001615330515065046,0.001619748448581368,0.001632724854359634,0.001681678032752793,0.001681678032752793,0.001691077527781684,0.001693679280308001,0.001717304541123761,0.001762437718019467,0.001836193693393892,0.001849042979525666,0.001850466600826215,0.001897263034190963,0.001908566809746642,0.001908566809746642,0.001946521398281333,0.001946521398281333,0.001953240397503532,0.001953240397503532,0.001968749511368946,0.001986176611711043,0.001986176611711043,0.001989162460415305,0.00203407028081538,0.002055242410669472,0.002055242410669472,0.002267924247161908,0.002267924247161908,0.002334607742120867,0.002334607742120867,0.002343948986994381,0.002415595210780518,0.002428743131036195,0.002428743131036195,0.002439868618816488,0.002481638625535113,0.00250176797050897,0.002502565763423053,0.002539691663578811,0.0025506106925679,0.002665516711800878,0.002671765500110548,0.002671765500110548,0.002702379328930651,0.002702379328930651,0.002721007127911129,0.002740281880749389,0.002740281880749389,0.002783630662331151,0.002811430089157377,0.002838173360744794,0.002885609859518302,0.002885609859518302,0.002892768698158465,0.003113312262886401,0.00312249921411428,0.00312249921411428,0.003192393846251505,0.003210333870540226,0.003277926779470305,0.003453201910876054,0.003455870039609966,0.003455870039609966,0.003510805066054018,0.003528508305586299,0.003549239319338777,0.003602661494241321,0.003602661494241321,0.003602661494241321,0.003651560340096738,0.003728011385835417,0.003741769438580605,0.003741769438580605,0.003741769438580605,0.00375662545086338,0.003791048887782826,0.003791048887782826,0.003791048887782826,0.003826604117998865,0.003929830525647058,0.003979814611532517,0.003979814611532517,0.004205549917677425,0.004205549917677425,0.004260645395984167,0.004279771087310983,0.004284453041531641,0.004310227735095518,0.00434206060772418,0.004343826470299544,0.004343826470299544,0.004348232803458339,0.004370136259245415,0.004386478111257473,0.004390850721247811,0.004513172214543498,0.004563845003044431,0.004599069104575581,0.004743341859877489,0.004750709468372125,0.004833615439206265,0.004833615439206265,0.004955308680416361,0.004984750540842261,0.004984750540842261,0.004984750540842261,0.004984750540842261,0.004984750540842261,0.005072807710817036,0.005115424168204322,0.005123045900744755,0.005145595220460069,0.005181625247133047,0.005188454147903406,0.0052033713804332,0.005212712977114634,0.005212712977114634,0.005277440815634748,0.00537382930232743,0.005375910017699308,0.005395920916941012,0.005395920916941012,0.005395920916941012,0.005461158770701342,0.005570700675395557,0.005570823343139918,0.005632000094983796,0.00567039340429702,0.00567039340429702,0.00567039340429702,0.00567039340429702,0.005701107830394124,0.005701107830394124,0.005755929883541781,0.005840213762828476,0.005973620180530739,0.005981763675729545,0.006025535529535778,0.006076160436662838,0.006105783432575482,0.006143556704807904,0.006185806465275502,0.006217682722779613,0.006217682722779613,0.006217682722779613,0.006224132515447945,0.006271345656195073,0.006271345656195073,0.006293286258811859,0.006401792248255956,0.006401792248255956,0.006401792248255956,0.006432177065940955,0.006514007060618473,0.006514007060618473,0.006714477007029634,0.006715048851373889,0.006715048851373889,0.006715048851373889,0.00678172719458647,0.006985180762761289,0.007036859179610429,0.007380892367881867,0.007380892367881867,0.007456981124846914,0.007525427427271207,0.007545452565438552,0.007655770012902227,0.007692442547052938,0.007804036075648809,0.007838142489473017,0.007838142489473017,0.007874352340210725,0.007888936493759413,0.007888936493759413,0.007897304993823857,0.007955288332544972,0.007955288332544972,0.007955288332544972,0.007959343726358085,0.008182361470169177,0.008182361470169177,0.008304193540142297,0.008304193540142297,0.008441199929644743,0.008441199929644743,0.008441199929644743,0.008494917353735114,0.008494917353735114,0.008552480526464627,0.008606436305682896,0.008606436305682896,0.008606436305682896,0.008698439003835748,0.008698439003835748,0.008738097307319144,0.008781599487927476,0.008791105647949602,0.008815020667196635,0.008894688884568668,0.00893757945421853,0.00911141240350441,0.009215264331404389,0.009241105834951982,0.009257337789382581,0.009333958370565922,0.009345865510120295,0.009346022523378322,0.00936875450947198,0.009455535373896855,0.009714731122999538,0.009714731122999538,0.01003750299097842,0.01004439349930355,0.01005185374244598,0.01007553828434842,0.01015361939320329,0.0102734739257248,0.01029775285351942,0.01063393376231978,0.01064166005616477,0.01076696621631664,0.01076696621631664,0.01076696621631664,0.0108037214083132,0.0108037214083132,0.0108037214083132,0.01085179765777241,0.01085777124275478,0.01094455017520965,0.01094455017520965,0.01094455017520965,0.01096868278091975,0.01135200103718867,0.01155535702407345,0.01155535702407345,0.01171530638683026,0.01171530638683026,0.01173056152988724,0.01173056152988724,0.01193045958775658,0.01193045958775658,0.01201418339104237,0.01207265786382464,0.01250611429344033,0.01254683585076597,0.01262827198282265,0.01290709584764856,0.01318734208001788,0.01318734208001788,0.01318734208001788,0.01318734208001788,0.0134100143762219,0.0134100143762219,0.0134100143762219,0.01364479444507,0.01399114854207364,0.01399114854207364,0.01404272950388235,0.01408316840200491,0.01420909529205453,0.01429138066147279,0.01439215927372259,0.01439215927372259,0.01439215927372259,0.01439215927372259,0.01456803141977059,0.01456803141977059,0.01465516714137502,0.01467637825096517,0.01475042226955353,0.01495904741644859,0.01499292486847147,0.01513385729782749,0.01529656016386181,0.01537811415524209,0.01547230961951843,0.01547230961951843,0.01547230961951843,0.01554918320819931,0.0155530883643205,0.0164501088625386,0.01655956953180234,0.01674209649180625,0.01701688032736872,0.01705528781056888,0.01725886221275164,0.01743927765799045,0.01795529531226275,0.01800790766045723,0.01800790766045723,0.01813005911393734,0.0181578552537623,0.0181578552537623,0.0181578552537623,0.0181578552537623,0.01828644995732353,0.01883075527252724,0.01898058592743149,0.01975511474027631,0.02027060110465451,0.02032407576280883,0.02063689527411858,0.02064677377050301,0.02075813559955721,0.02102242392860343,0.02102242392860343,0.02119436477659456,0.02123708522164878,0.02130336103810331,0.02156781629020513,0.02201114813853861,0.02210027396515757,0.02212067482548018,0.02258962561791546,0.02280049258812096,0.02320873455547414,0.02320873455547414,0.02341935652774486,0.02357031796397139,0.02360302560074276,0.02378741562630553,0.02393935931525603,0.02426895885171795,0.02483416824741422,0.02498287118746339,0.02510118381889531,0.02537779392372511,0.02585861272453888,0.0258745409666279,0.02596552720712328,0.02596552720712328,0.02596552720712328,0.02643252868959783,0.0266591917310129,0.02723220540645077,0.02726702419996919,0.02746988753204106,0.02765973198464793,0.02768562273230746,0.02791461718553226,0.02791461718553226,0.02798219749666007,0.02852349088415546,0.02879166548572178,0.02879166548572178,0.02948632159667793,0.02953605847677964,0.02997821364794581,0.03027946373729259,0.03049177839882571,0.03049177839882571,0.03075672063893783,0.0308973916016977,0.03093991128794154,0.03093991128794154,0.03097591182508881,0.03116040622783843,0.03116040622783843,0.03116040622783843,0.03124879145505799,0.03157402971305961,0.03176103549448504,0.03176103549448504,0.03176103549448504,0.03211031712439384,0.03292176620517985,0.03292176620517985,0.03322781787034971,0.03332703286865551,0.03395505128859692,0.03395505128859692,0.03398965605304637,0.03454634326549508,0.03502947517105422,0.03531411842871063,0.0358546051389272,0.03686789507203459,0.03721706125970052,0.03721706125970052,0.03736257762654631,0.03736257762654631,0.03840252783357078,0.03890630342041299,0.03927518025581836,0.03927518025581836,0.03934823176017108,0.03941337504666236,0.03941337504666236,0.0395006409405954,0.0401657231751931,0.04021076912916141,0.04031749066386764,0.04052584298194103,0.04066732788618611,0.04160003718062671,0.04200891794317795,0.04200891794317795,0.04242914403207197,0.04277446217934697,0.04279687507469035,0.04300128041916887,0.04303057314179462,0.04314206773009972,0.04322564253248343,0.0433092513576261,0.04372320487856923,0.04440879398698096,0.04480332677973348,0.04487073457203942,0.04487073457203942,0.04487073457203942,0.04549945567268874,0.04569010446143544,0.04570839079054573,0.04616022846443734,0.04635798031849017,0.04635798031849017,0.04635798031849017,0.04645392461640874,0.04702357387896011,0.04752976696694616,0.0477697173675,0.04823351316899634,0.04840918500801895,0.0486256244405005,0.04866113448189914,0.04900170241897372,0.04945686386525453,0.04978648403389641,0.04978648403389641],["22:30396941-30425303:+","9:33441156-33447596:-","1:207018522-207032756:-","5:36246913-36302114:-","14:35401513-35404749:-","19:15615218-15630639:+","6:35573585-35728583:-","2:191245185-191425389:+","1:241532134-241595642:+","8:127735434-127742951:+","X:10015254-10144474:+","1:221701424-221742089:-","21:30214006-30216097:-","4:69199951-69214748:-","22:45920366-45977162:-","7:146116002-148420998:+","12:101696947-101744140:+","16:54283304-54286787:-","4:155854738-155920406:+","6:25279078-25620530:+","19:4969113-5153598:+","4:69280475-69295050:+","20:1266280-1309328:+","22:31618491-31662432:-","17:68267026-68291267:-","5:36192589-36242279:-","3:189631389-189897276:+","11:45647689-45665622:-","14:64465499-64474503:+","4:74933095-75050115:+","2:31334321-31414742:-","18:26226445-26391685:+","16:67430652-67437553:+","3:183253253-183298504:+","15:38252836-38357249:+","6:105273218-105454062:-","18:8705661-8832778:+","11:60842113-60851081:+","7:2728105-2844308:-","22:33162226-33922841:-","5:168291645-168472303:+","12:47075707-47079951:-","21:26835755-26845409:-","7:87345664-87399794:+","12:48995149-48999375:-","1:65792514-66374579:+","10:111077029-111080907:+","7:99966720-99976042:-","18:58044226-58401540:+","X:67544021-67730619:+","20:58228940-58309451:-","9:137299631-137302271:-","16:22297375-22335101:+","10:102152176-102163871:+","5:14581792-14616180:+","7:99980762-99987535:+","5:76403285-76708132:+","10:68956170-68985068:+","17:4538897-4555386:-","18:9475009-9538114:+","2:230165186-230225729:-","21:36156782-36294274:+","12:82686880-83134870:+","5:169637268-170083382:+","11:120236640-120319945:+","8:89901849-89927888:+","4:69408828-69423164:+","10:100987367-100994403:+","X:7147237-7804358:+","3:188947214-189325304:+","2:96143169-96145440:-","17:47951967-47957883:-","11:7513298-7657127:+","4:152320544-152536092:-","5:128257909-128659185:-","8:8317736-8386439:-","3:126342635-126357408:-","1:169105697-169132722:+","8:123178960-123210079:+","X:154762742-154777689:+","5:74627406-74641424:-","1:2403964-2413797:-","1:117929720-117966543:+","2:136114349-136118149:-","16:52436417-52547802:-","20:40685848-40689236:-","15:55202966-55319113:-","8:27311482-27459391:+","7:16791811-16833433:-","19:10106362-10112012:+","19:2476122-2478259:+","8:66429014-66430733:+","1:220690363-220699153:+","2:71064344-71079808:+","5:176383938-176388927:-","5:154190730-154420984:+","13:23328826-23433740:-","1:112911847-112957013:-","11:69475567-69481545:-","2:99545419-100192428:-","1:109616104-109632053:+","4:159665833-159777828:-","2:19910260-19990131:-","4:152936352-152979696:+","21:42062959-42143453:+","11:129815848-129860003:+","4:69389492-69402081:+","17:79932343-80035872:-","11:107790991-107928293:-","20:34704339-34713439:+","8:94925972-94949378:-","8:144262045-144291438:-","7:101287482-101289771:-","20:24931840-24932983:+","2:10439968-10448327:-","2:171522247-171558129:+","1:54849627-54887195:-","17:57085092-57121346:+","X:133300103-133415489:-","3:172505508-172523475:-","19:15564074-15584709:+","22:23536881-23547797:+","8:109573978-109691791:-","X:11117651-11665920:-","17:8140472-8156506:-","3:111732497-111976517:+","11:19350724-20121601:+","4:4267701-4290154:-","9:33461353-33473930:-","12:101568353-101686028:+","3:51933429-51941904:-","2:149038107-149215262:+","2:118088452-118110997:+","9:38392702-38398661:+","3:8501807-8574668:+","8:37858618-37899497:-","12:131929200-131945896:+","2:45005182-45009452:-","10:113550837-113589602:+","22:42269703-42279534:+","1:58780791-58784047:-","1:182378098-182392206:-","11:34621093-34663288:+","8:98117293-98159835:+","13:23979805-24307074:+","5:149730298-149855022:+","17:39737927-39747291:+","1:43970000-43973369:+","X:105822539-105958610:+","6:150865549-151101887:+","2:218607855-218637184:+","X:110002631-110182734:+","6:36676460-36687337:+","2:8678845-8684461:+","13:42040036-42256578:+","1:56645314-56715335:+","17:5282265-5386340:+","17:49289206-49362473:-","2:226800146-226813741:+","6:41921499-41933046:+","11:86791059-86952910:+","6:81491439-81752774:-","6:139371807-139374648:-","3:121894401-121944188:+","5:133196455-133612541:-","15:40087890-40108892:-","X:74421493-74533917:+","17:64953683-64956546:+","2:15940550-15947007:+","20:58309715-58367507:+","9:136401922-136410614:-","1:147155106-147172550:-","5:149551947-149634968:+","4:37453925-37623495:+","3:124080023-124726325:+","3:121835183-121886526:+","3:101779202-101828231:+","6:56457987-56954649:-","17:82031678-82034204:+","5:173314713-173329503:-","2:28894667-28948219:+","6:129576132-129710177:-","3:156673235-156706770:+","11:60890547-60906585:-","7:41960949-42237870:-","3:105366909-105576900:+","7:50458436-50565405:-","4:73431138-73456174:+","X:132626016-132961395:-","2:12716889-12742734:+","6:7389793-7418037:+","11:46277662-46321409:+","17:40922700-40937646:-","9:98069275-98119222:-","22:31122731-31134697:+","10:3066333-3137718:+","1:67685201-67688334:+","1:11054584-11060020:-","10:96189171-96271587:-","11:113387779-113400416:+","18:54717860-54895516:+","22:26521996-26596717:-","17:39687914-39730426:+","20:62410237-62427539:-","11:95089846-95132645:+","1:229256892-229271242:-","11:76782251-76798153:+","3:14810853-14934571:+","3:38103129-38137242:-","2:9106593-9109865:+","5:16473038-16617058:-","12:53046969-53064379:+","10:28677510-28682932:+","1:232397965-232630571:-","X:107628428-107651993:+","16:71565660-71577092:-","16:15949577-16143074:+","X:109624244-109733403:-","16:83789967-83807834:-","2:112541915-112579818:+","5:71587288-71658706:+","X:132203024-132219480:-","5:113022099-113488823:-","18:23134564-23260467:+","14:65410592-65744121:+","1:46040140-46133036:-","10:71212570-71302864:+","12:110468415-110490385:+","17:2322396-2336657:-","12:29149103-29340980:+","11:113314579-113383544:+","1:25430858-25500209:+","11:35618460-35620865:+","10:113751262-113783429:+","16:1612337-1677908:+","9:107484852-107490482:-","19:36259540-36331770:-","5:132369710-132395613:+","5:159157409-159210053:-","1:89106132-89150456:-","14:91271323-91417844:-","5:154887411-154938211:-","12:31280584-31327058:-","7:106248298-106286326:-","4:184755595-184826818:-","X:108719949-108724944:-","11:47214465-47239217:+","2:230034982-230068993:-","13:46553168-46753040:+","15:63907036-64072033:-","7:105439661-105522271:-","1:161184302-161199054:-","2:86213993-86338083:-","4:145098288-145129524:+","3:120684938-120742993:-","5:90458209-90474771:-","19:41386374-41397359:-","9:115019575-115118257:-","18:61808067-61894247:-","13:23570370-23676104:+","16:54930865-54934485:+","10:102132994-102150333:+","8:74234700-74321540:-","2:68157888-68176238:+","16:80597907-80804598:-","3:58008400-58172251:+","11:47177522-47186443:-","12:7188685-7218574:+","22:42553617-42574382:+","22:30425623-30429054:+","20:38419638-38435409:-","17:72121020-72126416:+","1:229271062-229305894:+","2:197705369-197708395:+","1:203340628-203351758:-","19:43460787-43465608:-","5:172325000-172454525:-","2:71178587-71178773:+","1:203626787-203744081:+","16:72782885-73891871:-","6:35213956-35253079:+","15:56244009-56445997:+","11:11270877-11622005:-","16:4371848-4383538:+","15:64963022-64990310:-","14:95181940-95319908:-","2:150468195-150539011:-","12:54369133-54391298:-","11:65890673-65891635:+","1:236215101-236281958:-","7:43582758-43650713:+","15:88621337-88632281:+","12:39293228-39443390:-","11:104942866-104969366:-","20:63540810-63547504:-","5:78485215-78770021:-","2:214931542-215138626:-","1:178725147-178921841:+","6:57172326-57189833:+","14:70809841-70815994:+","1:169849631-169894267:-","1:16124337-16156069:-","11:62832342-62841809:-","1:207034366-207081024:+","7:99992397-99993050:+","1:24502351-24541040:+","1:220528136-220664461:+","12:109088188-109093631:-","3:154121003-154257827:+","3:187721377-187745725:-","2:130342877-130347967:+","19:10718079-10833488:+","2:8666636-8681864:-","5:128083766-128189677:+","17:64039142-64130819:-","3:192796815-192917856:-","16:50407184-50546347:+","20:49632945-49713878:-","10:94333226-94362959:-","19:19033575-19060311:+","20:2652593-2658393:+","2:186694060-186765959:+","11:57406954-57427580:-","7:157010805-157016426:+","3:180323107-180325059:+","13:26044597-26052016:-","6:143422832-143450695:-","3:121769761-121835079:-","19:41425359-41428730:-","1:40783787-40840452:+","22:21628089-21630064:-","1:117143587-117210960:-","7:139452690-139483673:-","X:37349309-37457292:+","10:103396626-103446294:+","16:69337996-69343106:+","7:23562459-23564261:-","9:2720469-2844095:-","18:58671465-58754477:+","3:50606489-50611774:-","2:138501770-138573547:+","8:133237175-133302022:-","10:36521721-36524234:-","2:221418027-221574202:-","12:124946825-124989131:-","1:21596215-21669363:-","3:183178043-183428778:-","20:833715-857463:+","22:43166622-43187134:-","17:64147227-64263260:-","17:76470891-76501790:-","5:65722205-65871725:+","11:83455012-85627922:-","12:48857145-48865870:-","3:52681156-52694497:+","1:233614106-233672514:+","3:50287732-50292918:-","4:39498755-39528311:-","6:93240020-93419559:-","2:121726945-121736911:-","4:77157207-77433388:+","21:34787801-36004667:-","12:54395261-54419266:-","2:238238267-238240662:-","6:51615299-52087613:-","12:93569814-93583487:+","7:94509219-94557019:+","3:194685883-194689037:+","4:56393362-56435615:-","14:32329298-32837684:+","3:143971823-144048719:+","10:79347469-79355334:+","22:27748277-27801756:-","1:19596979-19658456:+","9:12775020-12823060:+","18:10666483-11149569:-","2:5932543-6003510:+","1:169463909-169485944:-","12:48965340-48971735:-","17:37084992-37406836:-","19:51889235-51905040:-","14:36657568-36679362:+","3:123067025-123225227:+","11:123723914-123741675:-","21:43659560-43696079:+","8:101492439-101669726:+","14:61695513-61748259:+","16:48081006-48156018:-","1:229626247-229660200:+","1:246775880-246792385:+","11:47239302-47248906:-","21:15928296-16645467:+","8:144078648-144080648:+","13:27424619-27435823:+","12:89347235-89352501:-","17:3860315-3894891:-","1:151540305-151583583:+","1:19682213-19799945:-","5:83471618-83582303:+","13:19304593-19305625:-","1:182598623-182604389:-","10:101781325-101783413:-","X:44523639-44542859:-","5:38474668-38608354:-","3:9945542-9952408:-","5:151100706-151157785:-","3:184561785-184582408:+","1:203305491-203309602:+","19:45769709-45782552:-","20:21499261-21502934:-","7:99923266-99929620:-","21:32311018-32393012:-","1:167916675-167937072:-","15:63042632-63071915:+","12:122726076-122730844:-","2:226731317-226799759:-","21:6111131-6123778:+","17:3923870-3964464:-","16:79585843-79600737:-","20:10637684-10673999:-","18:32124877-32470882:-","16:4257186-4273075:-","5:112419583-112420978:+","17:54968727-55173632:+","5:177446445-177456286:+","12:123671113-123708399:+","1:42925353-42958893:-","3:12484432-12539623:+","9:76419850-76651203:+","17:80134369-80147151:-","4:155924118-155953866:-","8:11676959-11760002:+","1:53242364-53328469:-","11:134150113-134225454:-","1:20482391-20486210:-","X:153687926-153696588:+","17:62627670-62693597:+","X:64185117-64205708:-","17:36534987-36539310:+","15:51723011-51751585:-","2:202874261-203014798:-","1:65147549-65232145:+","1:180632022-180890279:+","21:44107373-44131181:+","10:19816239-20289856:+","12:121712752-121783001:+","16:47077703-47143945:-","11:27040725-27127809:+","20:1180561-1185415:-","11:115169218-115504957:-","7:149764182-149773588:-","4:37590800-37686376:-","13:85792790-85806683:-","6:10694972-10709782:+","17:77086716-77099902:+","11:86945679-86955395:-","11:45921621-45929096:+","X:110346703-110346909:-","12:12891559-12917937:+","3:27372721-27484420:-","3:141228726-141367753:+","5:80111651-80256048:-","12:57768471-57772119:-","5:115520908-115544775:-","14:28264390-28265974:+","15:60488284-61229302:-","17:38869859-38921770:+","1:3857267-3885429:+","2:24491914-24770702:+","8:123497889-123541206:-","22:31281594-31292534:-","14:37595847-38041442:+","16:5084284-5097795:-","1:165827614-165911618:+","11:73264505-73298625:+","11:47980558-48170841:+","3:107220744-107348464:+","11:76800364-76804555:+","10:26697701-26746798:+","9:4792944-4885917:+","5:178149463-178153894:-","21:5130871-5154658:-","8:58552924-58582859:+","16:88706483-88715396:+","11:10858179-10908972:+","16:15395754-15413271:+","2:195575977-195737702:+","5:139274102-139284899:+","X:38801432-38806537:+","12:69825227-69959097:+","6:12290361-12297194:+","4:75642782-75724525:-","17:38530031-38605952:-","10:122560665-122643736:+","2:8852690-9003709:-","16:3651639-3717553:-","14:29576479-30191898:-","6:3585101-3720081:-","15:39934115-40035591:+","11:68754620-68844410:-","3:100492619-100577444:+","4:88709298-88730103:+","1:116909916-116990353:+","7:107044705-107161811:+","9:137062127-137070557:-","10:5638867-5666595:-","4:168497052-168928457:+","2:109614364-109618990:+","12:52897187-52949954:-","5:116443555-116574823:-","5:88717117-88904257:-","7:44983023-44986961:-","6:10492223-10629368:+","14:22836560-22849027:+","12:112906783-112933222:+","1:22117313-22143969:-","1:41361922-41383590:+","7:94394895-94431227:+","5:112876385-112922289:-","14:103529196-103537073:+","16:30423615-30430030:-","11:17208153-17349980:+","17:75272981-75289510:-","2:32946953-33399509:+","7:111726110-112206407:-","11:85694224-85811159:-","9:99104038-99154192:+","19:50797704-50804929:-","17:50055968-50090481:+","1:156594301-156601496:-","13:97221434-97394120:+","19:39825350-39834201:-","10:120457227-120592065:+","7:107743949-107761667:+","7:93099513-93118023:-","2:189560590-189583758:-","20:34280268-34311802:-","14:97924637-97978124:-","21:42843094-42879568:-","15:43510958-43531620:+","6:16299112-16761491:-","1:154405193-154469450:+","8:37695782-37700019:+","18:23992773-24135610:+","2:231453531-231483641:-","12:46383652-46876784:+","1:116574399-116667755:-","7:101308270-101314800:+","14:56117814-56301524:+","X:22032325-22251310:+","15:90352284-90369146:+","19:34404399-34426168:+","14:37589552-37596059:-","17:47844908-47855874:-","12:125138245-125141711:+","3:177019340-177228000:-","2:191678068-191741097:+","3:149229356-149291800:-","5:176526712-176538025:-","6:108986437-109094819:-","14:105140982-105168824:-","2:87454781-87636740:+","2:237324003-237414207:-","17:28721487-28722877:-","2:218658764-218663443:+","1:84398532-84415018:+","4:105894775-106004027:+","5:79689836-79800240:+","16:87326987-87392142:-","X:78129748-78139650:-","16:74999024-75110994:+","21:33485530-33491716:-","15:31326835-31435665:+","5:1877413-1887236:-","4:103105349-103198445:-","13:73054976-73077541:+","2:173075435-173268015:+","11:62337448-62393412:+","15:38865322-39427195:-","1:151762899-151771334:+","6:4994717-5004063:-","11:64305497-64316743:+","1:36224432-36305357:+","8:85177154-85217158:+","6:122789049-122809720:+","2:96536743-96552634:+","11:113368478-113369117:+","5:154325568-154445850:-","5:132294394-132344190:+","5:111091716-111130502:+","2:26308547-26395891:+","19:15049480-15058293:+","19:751112-764318:+","20:58069054-58072070:-","21:43414483-43427131:-","5:150000046-150053142:+","1:40132764-40133448:+","16:69741871-69754926:-","6:31676684-31684040:-","6:3268962-3457022:-","X:120426148-120469365:-","4:78551747-78610451:+","3:3799431-3849834:+","1:27098809-27166981:-","2:232606057-232682780:+","9:137582081-137590512:-","11:74988279-75018893:+","7:47275154-47582558:-","6:34018643-34155622:-","12:47963569-47968878:+","6:148272304-148552048:+","14:69484692-69531551:+","17:13755574-14069495:-","X:46912276-47061242:+","8:27284886-27311272:-","5:60743563-60744970:-","3:149284782-149333653:+","16:48166910-48247568:-","3:114005833-114088422:+"],["protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_unprocessed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_unprocessed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","unprocessed_pseudogene","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_unprocessed_pseudogene","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_unprocessed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","unprocessed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","unprocessed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","transcribed_unitary_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","TEC","protein_coding","protein_coding","transcribed_unitary_pseudogene","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","lncRNA","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","processed_pseudogene","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","lncRNA","protein_coding","protein_coding","processed_pseudogene","lncRNA","protein_coding","protein_coding"],[[23541],[360],[79098],[202151],[4792],[11283],[2289],[4430],[8564],[4609],[55841],[11221],[9073],[10720],[7477],[26047],[56994],[79191],[6999],[55604],[23030],[54490],[9751],[23761],[9120],[133686],[8626],[8534],[9495],[25849],[7498],[6875],[3291],[84002],[161742],[5550],[23255],[79080],[2768],[9215],[23286],[347902],[9510],[54677],[23109],[5142],[150],[563],[23327],[367],[null],[441478],[55718],[9221],[54491],[646282],[10788],[9188],[10514],[10928],[3431],[9980],[160335],[1794],[25833],[734],[null],[56652],[412],[285386],[1844],[79170],[8495],[55294],[2201],[157285],[28999],[481],[84985],[1736],[8507],[5192],[10885],[7852],[27324],[9935],[5873],[2185],[10551],[56342],[4616],[23212],[79762],[55577],[51491],[55568],[26278],[6566],[105379407],[3899],[271],[null],[57539],[85462],[89766],[120224],[null],[125058],[54733],[58476],[94241],[23246],[null],[null],[4953],[79901],[1718],[8165],[2239],[8743],[null],[null],[55638],[395],[5187],[90102],[89797],[55646],[65083],[4604],[9136],[130576],[51141],[219],[29995],[80223],[80324],[10736],[3026],[null],[3725],[2752],[26298],[10940],[221178],[133522],[2886],[1802],[203447],[25902],[84812],[84187],[1026],[3398],[160851],[5563],[9135],[22834],[null],[705],[11098],[55603],[10370],[6565],[23105],[90427],[6567],[null],[4613],[57403],[10807],[5565],[389337],[55286],[8997],[55840],[91775],[667],[5881],[8614],[23160],[93663],[25976],[27339],[2737],[214],[1644],[174],[90161],[28951],[83732],[90993],[25984],[9830],[27124],[5214],[1647],[6723],[29760],[255239],[5874],[8459],[2064],[140893],[23052],[105373159],[25987],[152273],[30],[null],[54463],[23371],[25805],[57568],[5631],[6898],[4363],[2182],[102724163],[84172],[64087],[57826],[4163],[91768],[2530],[8503],[219699],[29902],[55720],[55711],[54970],[55219],[24147],[79949],[57585],[9314],[100506930],[6584],[153830],[2634],[440193],[25929],[58516],[10135],[2180],[null],[1643],[151473],[23143],[23604],[54517],[9507],[65055],[6059],[285282],[153364],[56915],[3371],[220441],[55504],[10265],[23082],[56704],[56902],[124359],[2317],[29763],[5830],[253190],[51537],[388796],[6662],[5867],[92935],[2331],[27076],[285590],[null],[493],[463],[222663],[374618],[374378],[114990],[51324],[79789],[390],[25946],[11007],[56605],[9263],[64782],[55605],[837],[6725],[10184],[26154],[55103],[9532],[100506411],[57147],[1969],[54663],[5208],[null],[11123],[4139],[121642],[26084],[604],[92856],[1785],[null],[6558],[2081],[151963],[null],[9334],[64318],[93436],[10528],[165215],[29015],[645249],[null],[387914],[134637],[9657],[374907],[9132],[150223],[79679],[346689],[5638],[22984],[51388],[null],[9933],[10892],[1154],[339745],[10397],[null],[2043],[57647],[5909],[23101],[83541],[23170],[55852],[79651],[57486],[1740],[27289],[26354],[3775],[7866],[7358],[2045],[84365],[901],[861,100506403],[3678],[55502],[5314],[8835],[64921],[131583],[5471],[9472],[205428],[10105],[4330],[4681,100532736],[286343],[63895],[null],[10560],[7480],[31],[65251],[5083],[10954],[7753],[23076],[79977],[3091],[94160],[9816],[null],[53],[null],[54512],[2971],[1848],[84254],[7286],[255104],[1462],[null],[6004],[10360],[139341],[3977],[285368],[309],[2049],[7832],[1760],[null],[349149],[9875],[25874],[7168],[27198],[3667],[102724428],[489],[4094],[182],[64762],[7023],[54508],[252983],[80758],[79867],[6513],[80746],[2650],[9775],[1519],[2626],[7804],[23310],[55450],[6535],[9902],[139285],[284098],[256586],[55759],[205],[9213],[5822],[84898],[144404],[81831],[8424],[55321],[23705],[168544],[768211],[84189],[55003],[654434],[8322],[120071],[null],[9052],[9497],[92370],[256987],[4234],[56929],[null],[6095],[3927],[1677],[8648],[114907],[113791],[319089],[196483],[7371],[5031],[5795],[344595],[null],[23590],[10171],[55651],[102724159],[6386],[348180],[729013],[255027],[57181],[null],[58526],[196446],[1906],[9908],[80725],[1755],[129642],[10131],[5587],[100507336],[440275],[1374],[55076],[285512],[5738],[5577],[89958],[79754],[23022],[65124],[3856],[57556],[4208],[285958],[2651],[4323],[4938],[54361],[100132074],[1278],[7905],[115708],[79077],[4925],[60386],[4052],[9732],[54843],[7046],[84798],[3675],[54865],[10150],[9149],[196051],[79872],[54809],[30061],[191],[388011],[10785],[4130],[6310],[3570],[80139],[125488],[4691],[null],[3321],[null],[57161],[5251],[342132],[84306],[3169],[80320],[null],[79718],[64859],[null],[22838],[27244],[3714],[112597],[1293],[null],[617],[58511],[255743],[202333],[79791],[51616],[84937],[54943],[51621],[50805],[1062],[688],[51776],[80150],[null],[51686],[10799],[2101],[9967],[1875],[10924],[10865],[null],[386627],[6583],[134430],[85465],[23581],[126353],[null],[150094],[22993],[null],[28987],[80741],[63027],[3920],[306],[57633],[6548],[80303],[116225],[10825],[64759],[2914],[79022],[23328],[400224],[100874058],[9767],[23087],[null],[null],[85320],[79691]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>gene_id<\/th>\n      <th>gene_name<\/th>\n      <th>logCPM<\/th>\n      <th>logFC<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n      <th>location<\/th>\n      <th>gene_biotype<\/th>\n      <th>entrezid<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render"],"jsHooks":[]}</script>
<div id="top-de-genes" class="section level3">
<h3>Top DE Genes</h3>
<p>logCPM values for the most up and down-regulated genes were also
inspected using simple boxplots.</p>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotTopUp-1.png" alt="*Top ranked up-regulated genes*" width="960" />
<p class="caption">
<em>Top ranked up-regulated genes</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotTopUp-1">
Past versions of plotTopUp-1.png
</button>
</p>
<div id="fig-plotTopUp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/838f1b8754b7ac9bd880e7dd2921a65bbb8d1e99/docs/figure/dge_analysis.Rmd/plotTopUp-1.png" target="_blank">838f1b8</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotTopUp-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotTopDown-1.png" alt="*Top ranked down-regulated genes*" width="960" />
<p class="caption">
<em>Top ranked down-regulated genes</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotTopDown-1">
Past versions of plotTopDown-1.png
</button>
</p>
<div id="fig-plotTopDown-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/838f1b8754b7ac9bd880e7dd2921a65bbb8d1e99/docs/figure/dge_analysis.Rmd/plotTopDown-1.png" target="_blank">838f1b8</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-08-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotTopDown-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="checks-for-bias" class="section level2">
<h2>Checks For Bias</h2>
<p>Using the sign of logFC to indicate direction, and -<span
class="math inline">\(\log_{10} p\)</span> as the ranking statistic,
bias was also checked within the set of results, using GC content and
gene length as potential sources of bias amongst the set of DE
genes.</p>
<pre class="r"><code>a &lt;- topTable %&gt;%
  ggplot(aes(signedRank, gc_content)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = mean(topTable$gc_content)) +
  labs(
    x = &quot;Rank&quot;, y = &quot;% GC Content&quot;
  ) +
  scale_colour_manual(values = c(&quot;grey50&quot;, &quot;red&quot;)) + 
  theme(legend.position = &quot;none&quot;)
b &lt;- topTable %&gt;%
    ggplot(aes(gc_content, stat(density))) +
    geom_density(colour = &quot;grey50&quot;) +
    geom_vline(xintercept = mean(topTable$gc_content)) +
    coord_flip() + 
    theme(
      axis.text = element_blank(),
      axis.title = element_blank(),
      axis.ticks = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
c &lt;- topTable %&gt;%
  ggplot(aes(signedRank, ave_tx_len)) +
  geom_point(aes(colour = DE), alpha = 0.4) +
  geom_smooth(se = FALSE) +
  geom_hline(yintercept = 10^mean(log10(topTable$ave_tx_len))) +
  labs(
    x = &quot;Rank&quot;, y = &quot;Gene Length (nt)&quot;
  ) +
  scale_y_log10(label = comma) +
  scale_colour_manual(values = c(&quot;grey50&quot;, &quot;red&quot;)) + 
  theme(legend.position = &quot;none&quot;)
d &lt;- topTable %&gt;%
  ggplot(aes(ave_tx_len, stat(density))) +
  geom_density(colour = &quot;grey50&quot;) +
  geom_vline(xintercept = 10^mean(log10(topTable$ave_tx_len))) +
  coord_flip() + 
  scale_x_log10() +
  theme(
    axis.text = element_blank(),
    axis.title = element_blank(),
    axis.ticks = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
plot_grid(
  plot_grid(ngsReports:::.emptyPlot(&quot;&quot;), a, b, rel_widths = c(0.35, 9, 1), align = &quot;h&quot;, nrow = 1),
  plot_grid(c, d, rel_widths = c(9, 1), align = &quot;h&quot;),
  nrow = 2,
  rel_heights = c(1, 1),
  labels = c(&quot;A&quot;, &quot;B&quot;),
  align = &quot;v&quot;,
  axis = &quot;l&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/dge_analysis.Rmd/plotBias-1.png" alt="*Genes ranked for signed differential expression shown against A) GC Content, and B) Gene Length. Ranks were assigned based on the ranking statistic R = -sign(logFC)*log~10~p, such that genes at the left are most down-regulated, whilst those at the right are the most up-regulated. Horizontal black lines indicate the overall average value, whilst blue curves indicate the localised GAM fit. A small positive bias was noted at both extremes for both GC content and Gene Length.*" width="960" />
<p class="caption">
<em>Genes ranked for signed differential expression shown against A) GC
Content, and B) Gene Length. Ranks were assigned based on the ranking
statistic R = -sign(logFC)</em>log<sub>10</sub>p, such that genes at the
left are most down-regulated, whilst those at the right are the most
up-regulated. Horizontal black lines indicate the overall average value,
whilst blue curves indicate the localised GAM fit. A small positive bias
was noted at both extremes for both GC content and Gene Length.*
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotBias-1">
Past versions of plotBias-1.png
</button>
</p>
<div id="fig-plotBias-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/blob/1d06dff563725688789f0fb9c68f321c15663e79/docs/figure/dge_analysis.Rmd/plotBias-1.png" target="_blank">1d06dff</a>
</td>
<td>
Steve Ped
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="exclusion-of-replicate-2" class="section level2">
<h2>Exclusion of Replicate 2</h2>
<p>Given the concerns raised by the above MA plot and the checks for
bias, the above analysis was repeated, excluding replicate 2 (not
shown). Whilst results were broadly comparable, this did not improve the
bias observed in the low-expressed genes. Instead, this bias was
attributed to genes expressed at low levels in only one of the treatment
groups.</p>
<p>Any enrichment analysis should incorporate a bias offset, such as
offered by the package <code>goseq</code>. Enrichment analysis using
ranked lists should be undertaken with caution.</p>
</div>
</div>
<div id="compare-degs-to-hub-genes-detected-in-the-pde-study"
class="section level1">
<h1>Compare DEGs to hub genes detected in the PDE study</h1>
<pre class="r"><code>PDE_DHT_hubGene &lt;- readRDS(&quot;~/WenjunLiu_Thesis_Chapter4/output/PDE_DHT_hubGene.rds&quot;)
lobular_diff &lt;- PDE_DHT_hubGene %&gt;%
  dplyr::filter(!Align) 
lobular_overlap &lt;- topTable %&gt;%
  dplyr::filter(DE) %&gt;%
  dplyr::select(
    gene_name, logFC, 
  ) %&gt;%
  left_join(
    lobular_diff 
  ) %&gt;%
  drop_na()</code></pre>
<p>88 hub genes potentially driving the perturbation of pathways in
DHT-treated IDC-like PDEs were loaded and compared against DEGs detected
in DHT-treated MDA-MB-453 cells. Only 4 overlapping genes were found.
All of them had directional opposing median ssFCs between IDC- and
ILC-like PDEs, but the direction of changes observed in MDA-MB-453 cells
aligned more to the DHT-treated IDC-like PDEs than ILC-like PDEs. Only
the HIF1A gene, which was upregulated in MDA-MB-453 cells, had
directional concordant chagne between the cell line and ILC-like PDEs
and opposite change to IDC-like PDEs.</p>
<pre class="r"><code>temp &lt;- lobular_overlap %&gt;%
  dplyr::select(
    Gene = gene_name, 
    `MDA-MB-453 logFC` = logFC, 
    `ILC-like PDE logFC` = `ILC-like`, 
    `IDC-like PDE logFC` = `IDC-like`
  ) %&gt;%
  mutate_if(is.numeric, round, 4) 
temp %&gt;%
  datatable() %&gt;%
  formatStyle(
    str_subset(colnames(temp), &quot;logFC&quot;), 
    color = styleInterval(0, c(&quot;blue&quot;, &quot;red&quot;))
  )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-855d8708d7e2ba0652dd" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-855d8708d7e2ba0652dd">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4"],["NFKBIA","PRKAA2","HIF1A","SLC2A1"],[1.8546,0.733,0.5607,0.482],[-0.1601,-0.0379,0.1866,-0.1873],[0.6803,1.1579,-0.117,0.9443]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>MDA-MB-453 logFC<\/th>\n      <th>ILC-like PDE logFC<\/th>\n      <th>IDC-like PDE logFC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<div id="data-export" class="section level2">
<h2>Data Export</h2>
<p>These above results were exported as the file
<code>MDA-MB-453_RNASeq.tsv</code>.</p>
<pre class="r"><code>topTable %&gt;%
  mutate(
    entrezid = vapply(entrezid, str_flatten, character(1), collapse = &quot;;&quot;)
  ) %&gt;%
  write_tsv(
    file = here::here(&quot;output/MDA-MB-453_RNASeq.tsv&quot;)
  )</code></pre>
<p>In addition, the final <code>DGEList</code> was exported for easier
integration with subsequent analyses, as were the CPM values for all
detected genes</p>
<pre class="r"><code>write_rds(
  dge,
  file = here::here(&quot;output/dge.rds&quot;)
)</code></pre>
<pre class="r"><code>2^logCPM %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  mutate(gene_name = dge$genes$gene_name) %&gt;%
  dplyr::select(starts_with(&quot;gene&quot;), everything()) %&gt;%
  write_tsv(&quot;output/MDA-MB-453_CPM.tsv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] DT_0.28                 ggrepel_0.9.3           cqn_1.46.0             
 [4] quantreg_5.95           SparseM_1.81            preprocessCore_1.62.1  
 [7] nor1mix_1.3-0           mclust_6.0.0            magrittr_2.0.3         
[10] ggfortify_0.4.16        cowplot_1.1.1           ensembldb_2.24.0       
[13] AnnotationFilter_1.24.0 GenomicFeatures_1.52.1  AnnotationDbi_1.62.2   
[16] Biobase_2.60.0          GenomicRanges_1.52.0    GenomeInfoDb_1.36.1    
[19] IRanges_2.34.1          S4Vectors_0.38.1        AnnotationHub_3.8.0    
[22] BiocFileCache_2.8.0     dbplyr_2.3.3            BiocGenerics_0.46.0    
[25] edgeR_3.42.4            limma_3.56.2            glue_1.6.2             
[28] pander_0.6.5            scales_1.2.1            yaml_2.3.7             
[31] lubridate_1.9.2         forcats_1.0.0           stringr_1.5.0          
[34] dplyr_1.1.2             purrr_1.0.1             readr_2.1.4            
[37] tidyr_1.3.0             tibble_3.2.1            ggplot2_3.4.2          
[40] tidyverse_2.0.0         workflowr_1.7.0        

loaded via a namespace (and not attached):
  [1] later_1.3.1                   BiocIO_1.10.0                
  [3] bitops_1.0-7                  filelock_1.0.2               
  [5] XML_3.99-0.14                 lifecycle_1.0.3              
  [7] rprojroot_2.0.3               processx_3.8.2               
  [9] lattice_0.21-8                vroom_1.6.3                  
 [11] MASS_7.3-60                   crosstalk_1.2.0              
 [13] plotly_4.10.2                 sass_0.4.6                   
 [15] rmarkdown_2.23                jquerylib_0.1.4              
 [17] httpuv_1.6.11                 DBI_1.1.3                    
 [19] zlibbioc_1.46.0               RCurl_1.98-1.12              
 [21] rappdirs_0.3.3                git2r_0.32.0                 
 [23] GenomeInfoDbData_1.2.10       MatrixModels_0.5-2           
 [25] codetools_0.2-19              DelayedArray_0.26.6          
 [27] xml2_1.3.5                    tidyselect_1.2.0             
 [29] farver_2.1.1                  matrixStats_1.0.0            
 [31] GenomicAlignments_1.36.0      jsonlite_1.8.7               
 [33] ellipsis_0.3.2                survival_3.5-5               
 [35] systemfonts_1.0.4             tools_4.3.0                  
 [37] progress_1.2.2                ragg_1.2.5                   
 [39] Rcpp_1.0.11                   gridExtra_2.3                
 [41] xfun_0.39                     here_1.0.1                   
 [43] mgcv_1.9-0                    MatrixGenerics_1.12.2        
 [45] withr_2.5.0                   BiocManager_1.30.21          
 [47] fastmap_1.1.1                 fansi_1.0.4                  
 [49] callr_3.7.3                   digest_0.6.33                
 [51] timechange_0.2.0              R6_2.5.1                     
 [53] mime_0.12                     textshaping_0.3.6            
 [55] colorspace_2.1-0              biomaRt_2.56.1               
 [57] RSQLite_2.3.1                 utf8_1.2.3                   
 [59] generics_0.1.3                data.table_1.14.8            
 [61] rtracklayer_1.60.0            prettyunits_1.1.1            
 [63] httr_1.4.6                    htmlwidgets_1.6.2            
 [65] S4Arrays_1.0.4                whisker_0.4.1                
 [67] pkgconfig_2.0.3               gtable_0.3.3                 
 [69] blob_1.2.4                    XVector_0.40.0               
 [71] htmltools_0.5.5               ProtGenerics_1.32.0          
 [73] png_0.1-8                     ngsReports_2.2.4             
 [75] ggdendro_0.1.23               knitr_1.43                   
 [77] rstudioapi_0.15.0             tzdb_0.4.0                   
 [79] rjson_0.2.21                  nlme_3.1-162                 
 [81] curl_5.0.1                    zoo_1.8-12                   
 [83] cachem_1.0.8                  BiocVersion_3.17.1           
 [85] parallel_4.3.0                restfulr_0.0.15              
 [87] pillar_1.9.0                  grid_4.3.0                   
 [89] vctrs_0.6.3                   promises_1.2.0.1             
 [91] xtable_1.8-4                  evaluate_0.21                
 [93] cli_3.6.1                     locfit_1.5-9.8               
 [95] compiler_4.3.0                Rsamtools_2.16.0             
 [97] rlang_1.1.1                   crayon_1.5.2                 
 [99] labeling_0.4.2                ps_1.7.5                     
[101] getPass_0.2-2                 fs_1.6.2                     
[103] stringi_1.7.12                viridisLite_0.4.2            
[105] BiocParallel_1.34.2           munsell_0.5.0                
[107] Biostrings_2.68.1             lazyeval_0.2.2               
[109] Matrix_1.6-0                  hms_1.1.3                    
[111] patchwork_1.1.2               bit64_4.0.5                  
[113] KEGGREST_1.40.0               statmod_1.5.0                
[115] shiny_1.7.4.1                 SummarizedExperiment_1.30.2  
[117] interactiveDisplayBase_1.38.0 highr_0.10                   
[119] memoise_2.0.1                 bslib_0.5.0                  
[121] bit_4.0.5                    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
