<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-09-19" />

<title>sSNAPPY</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MDA-MB-453_DHT-RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Workflow Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
    <li>
      <a href="dge_enrichment.html">Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">sSNAPPY</h1>
<h4 class="date">2023-09-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>MDA-MB-453_DHT-RNASeq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLMDAMB453DHTRNASeqtree9aaff4f369f404afe341b2b97af711cb4f52dc88targetblank9aaff4fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/tree/9aaff4f369f404afe341b2b97af711cb4f52dc88" target="_blank">9aaff4f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLMDAMB453DHTRNASeqtree9aaff4f369f404afe341b2b97af711cb4f52dc88targetblank9aaff4fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DRMCRL/MDA-MB-453_DHT-RNASeq/tree/9aaff4f369f404afe341b2b97af711cb4f52dc88" target="_blank">9aaff4f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/sPLS-DA.Rmd
    Untracked:  analysis/sSNAPPY.Rmd
    Untracked:  output/MDA453_permutedscore.rds

Unstaged changes:
    Modified:   analysis/dge_analysis.Rmd
    Modified:   output/MDA-MB-453_CPM.tsv
    Modified:   output/MDA-MB-453_RNASeq.tsv
    Modified:   output/dge.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(magrittr)
library(cowplot)
library(ggfortify)
library(cqn)
library(ggrepel)
library(DT)
library(randomcoloR)
library(ggforce)
library(sSNAPPY)
library(rvest)
library(xml2)
library(BSDA)
library(corrplot)</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<pre class="r"><code>if (!dir.exists(here::here(&quot;docs/assets&quot;))) dir.create(here::here(&quot;docs/assets&quot;))</code></pre>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- read_rds(here::here(&quot;output/genesGR.rds&quot;))</code></pre>
<p>Gene annotations were again loaded from Ensembl Release 101. The <a
href="qc_aligned.html#Annotation_Setup">previously defined</a>
<code>GenomicRanges</code> object containing GC content and Gene Length
was also loaded, containing information for 67,990 genes.</p>
</div>
<div id="read-in-data" class="section level2">
<h2>Read in data</h2>
<pre class="r"><code>diag_cols &lt;- readRDS(&quot;~/GSE800098/output/diag_cols.rds&quot;)</code></pre>
<p>The filtered and normalised <code>DGEList</code> was loaded in.</p>
<pre class="r"><code>dge &lt;- here::here(&quot;output/dge.rds&quot;) %&gt;%
  read_rds()
logCPM &lt;- dge$counts %&gt;%
  edgeR::cpm(log = TRUE)</code></pre>
</div>
</div>
<div id="retrieve-kegg-pathways" class="section level1">
<h1>Retrieve KEGG pathways</h1>
<p>To apply the <code>sSNAPPY</code> algorithm, pathway topologies of
KEGG pathways were firstly retrieved.</p>
<pre class="r"><code>gsTopology &lt;- retrieve_topology(database = &quot;kegg&quot;, species = &quot;hsapien&quot;)</code></pre>
<p>315 KEGG pathways were retrieved but as described in the PDE study,
irrelevant pathways are removed to keep only the 235 relevant
pathways.</p>
<pre class="r"><code>gsTokeep &lt;- readRDS(&quot;~/20131906_HickeyT_JC_NormalBreast/output/kg_gsTokeep.rds&quot;)
gsTopology &lt;- gsTopology %&gt;%
    .[names(.) %in% gsTokeep]</code></pre>
</div>
<div id="ssnappy" class="section level1">
<h1>sSNAPPY</h1>
<p>To compute the single-sample logFCs required, rownames of the logCPM
matrix need to be converted to <code>entrez id</code> first.</p>
<pre class="r"><code>rownames(logCPM) &lt;- mapIds(ensDb, rownames(logCPM), &quot;ENTREZID&quot;, keytype = &quot;GENEID&quot;)
# Remove genes that couldn&#39;t be matched to entrez IDs
logCPM &lt;- logCPM[!is.na(rownames(logCPM)),]</code></pre>
<p>Weighted single sample logFCs were computed from the logCPM
matrix.</p>
<pre class="r"><code>metadata &lt;- dge$samples %&gt;%
    dplyr::rename(treatment = treat) %&gt;%
    droplevels() %&gt;%
    mutate(
        treatment = relevel(treatment, ref = &quot;Vehicle&quot;)
    )
weightedFC &lt;- weight_ss_fc(logCPM, metadata, sampleColumn = &quot;name&quot;, 
                           treatColumn = &quot;treatment&quot;, 
                           groupBy = &quot;rep&quot;)</code></pre>
<p>Raw single sample perturbation scores for each treated sample each
KEGG pathway were calculated.</p>
<pre class="r"><code>gene_pert &lt;- sSNAPPY::raw_gene_pert(weightedFC$weighted_logFC, gsTopology)
ssPertScore &lt;- pathway_pert(gene_pert, weightedFC = weightedFC$weighted_logFC)</code></pre>
<p>Null distribution of perturbation scores were generated through
sample label permutations.</p>
<pre class="r"><code>set.seed(123)
permutedScore &lt;- generate_permuted_scores(
    logCPM,
    gsTopology = gsTopology,
    weight = weightedFC$weight
)
write_rds(permutedScore, here::here(&quot;output/MDA453_permutedscore.rds&quot;))

permutedScore &lt;- read_rds(here::here(&quot;output/MDA453_permutedscore.rds&quot;))</code></pre>
<p>Since the sample size is fairly short (ie. 12 samples), the maximum
number of possible permutation pairs is 132. The permuted distributions
don’t look very impressive.</p>
<pre class="r"><code>set.seed(123)
pl &lt;- permutedScore %&gt;%
    keep(~all(.!=0)) %&gt;%
    .[sample(seq_along(.), 6)] %&gt;%
    lapply(
      function(x){
        ggplot(mapping = aes(x)) + 
          geom_histogram() +
          xlab(&quot;Perturbation Score&quot;)
      }
    ) 
plot_grid(plotlist = pl, nrow = 2)</code></pre>
<p><img src="figure/sSNAPPY.Rmd/hist_pl-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="sample-wise" class="section level3">
<h3>Sample-wise</h3>
<p>Raw perturbation scores were converted to robust z-scores based on
the median and MAD derived from the null distributions. A permuted
p-value was computed for each single-sample perturbation scores and
corrected for multiple testings.</p>
<p>No pathway was defined to be significantly perturbed at individual
sample levels.</p>
<pre class="r"><code>normalisedScores &lt;- normalise_by_permu(permutedScore, ssPertScore)
normalisedScores %&gt;%
    dplyr::filter(adjPvalue &lt; 0.05)</code></pre>
<pre><code>[1] MAD       MEDIAN    gs_name   sample    score     robustZ   pvalue   
[8] adjPvalue
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="average-treatment-effect" class="section level3">
<h3>Average treatment effect</h3>
<p>Using the sample-level perturbation scores obtained above, we can
testing for group-level perturbation by applying t-tests across all
treated samples. Using a FDR cut-off of 0.05, only 2 pathways were
considered to be significantly perturbed</p>
<pre class="r"><code>z_matrix &lt;- normalisedScores %&gt;%
  dplyr::select(robustZ, gs_name, sample) %&gt;%
  # dplyr::filter(sample != &quot;MDA2-D&quot;) %&gt;%
  pivot_wider(names_from = &quot;sample&quot;, values_from = &quot;robustZ&quot;) %&gt;%
  column_to_rownames(&quot;gs_name&quot;) %&gt;%
  as.matrix()
sapply(rownames(z_matrix), function(x){
  test &lt;- t.test(z_matrix[x,]) 
  data.frame(
    gs_name = x, 
    pvalue = test$p.value, 
    t_statistic = test$statistic
  )
}, simplify = FALSE) %&gt;%
  bind_rows() %&gt;%
  mutate(FDR = p.adjust(pvalue, &quot;fdr&quot;)) %&gt;%
  dplyr::filter(
    FDR &lt; 0.05
  )</code></pre>
<pre><code>                      gs_name       pvalue t_statistic        FDR
t...1 kegg.Huntington disease 0.0001847496    9.839290 0.01563088
t...2  kegg.Viral myocarditis 0.0002155983    9.526681 0.01563088</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] corrplot_0.92           BSDA_1.2.1              lattice_0.21-8         
 [4] xml2_1.3.5              rvest_1.0.3             sSNAPPY_1.5.1          
 [7] ggforce_0.4.1           randomcoloR_1.1.0.1     DT_0.28                
[10] ggrepel_0.9.3           cqn_1.46.0              quantreg_5.95          
[13] SparseM_1.81            preprocessCore_1.62.1   nor1mix_1.3-0          
[16] mclust_6.0.0            ggfortify_0.4.16        cowplot_1.1.1          
[19] magrittr_2.0.3          ensembldb_2.24.0        AnnotationFilter_1.24.0
[22] GenomicFeatures_1.52.1  AnnotationDbi_1.62.2    Biobase_2.60.0         
[25] GenomicRanges_1.52.0    GenomeInfoDb_1.36.1     IRanges_2.34.1         
[28] S4Vectors_0.38.1        AnnotationHub_3.8.0     BiocFileCache_2.8.0    
[31] dbplyr_2.3.3            BiocGenerics_0.46.0     edgeR_3.42.4           
[34] limma_3.56.2            glue_1.6.2              pander_0.6.5           
[37] scales_1.2.1            yaml_2.3.7              lubridate_1.9.2        
[40] forcats_1.0.0           stringr_1.5.0           dplyr_1.1.2            
[43] purrr_1.0.1             readr_2.1.4             tidyr_1.3.0            
[46] tibble_3.2.1            ggplot2_3.4.2           tidyverse_2.0.0        
[49] workflowr_1.7.0        

loaded via a namespace (and not attached):
  [1] later_1.3.1                   BiocIO_1.10.0                
  [3] bitops_1.0-7                  filelock_1.0.2               
  [5] polyclip_1.10-4               graph_1.78.0                 
  [7] XML_3.99-0.14                 lifecycle_1.0.3              
  [9] rprojroot_2.0.3               processx_3.8.2               
 [11] MASS_7.3-60                   sass_0.4.6                   
 [13] rmarkdown_2.23                jquerylib_0.1.4              
 [15] httpuv_1.6.11                 RColorBrewer_1.1-3           
 [17] DBI_1.1.3                     zlibbioc_1.46.0              
 [19] Rtsne_0.16                    ggraph_2.1.0                 
 [21] RCurl_1.98-1.12               tweenr_2.0.2                 
 [23] rappdirs_0.3.3                git2r_0.32.0                 
 [25] GenomeInfoDbData_1.2.10       pheatmap_1.0.12              
 [27] MatrixModels_0.5-2            codetools_0.2-19             
 [29] DelayedArray_0.26.6           tidyselect_1.2.0             
 [31] farver_2.1.1                  viridis_0.6.4                
 [33] matrixStats_1.0.0             GenomicAlignments_1.36.0     
 [35] jsonlite_1.8.7                e1071_1.7-13                 
 [37] ellipsis_0.3.2                tidygraph_1.2.3              
 [39] survival_3.5-5                tools_4.3.0                  
 [41] progress_1.2.2                Rcpp_1.0.11                  
 [43] gridExtra_2.3                 here_1.0.1                   
 [45] xfun_0.39                     MatrixGenerics_1.12.2        
 [47] withr_2.5.0                   BiocManager_1.30.21          
 [49] fastmap_1.1.1                 fansi_1.0.4                  
 [51] callr_3.7.3                   digest_0.6.33                
 [53] timechange_0.2.0              R6_2.5.1                     
 [55] mime_0.12                     colorspace_2.1-0             
 [57] gtools_3.9.4                  biomaRt_2.56.1               
 [59] RSQLite_2.3.1                 utf8_1.2.3                   
 [61] generics_0.1.3                class_7.3-22                 
 [63] rtracklayer_1.60.0            prettyunits_1.1.1            
 [65] graphlayouts_1.0.0            httr_1.4.6                   
 [67] htmlwidgets_1.6.2             S4Arrays_1.0.4               
 [69] graphite_1.46.0               whisker_0.4.1                
 [71] pkgconfig_2.0.3               gtable_0.3.3                 
 [73] blob_1.2.4                    XVector_0.40.0               
 [75] htmltools_0.5.5               ProtGenerics_1.32.0          
 [77] png_0.1-8                     knitr_1.43                   
 [79] rstudioapi_0.15.0             reshape2_1.4.4               
 [81] tzdb_0.4.0                    rjson_0.2.21                 
 [83] curl_5.0.1                    org.Hs.eg.db_3.17.0          
 [85] proxy_0.4-27                  cachem_1.0.8                 
 [87] BiocVersion_3.17.1            parallel_4.3.0               
 [89] restfulr_0.0.15               pillar_1.9.0                 
 [91] grid_4.3.0                    vctrs_0.6.3                  
 [93] promises_1.2.0.1              xtable_1.8-4                 
 [95] cluster_2.1.4                 evaluate_0.21                
 [97] cli_3.6.1                     locfit_1.5-9.8               
 [99] compiler_4.3.0                Rsamtools_2.16.0             
[101] rlang_1.1.1                   crayon_1.5.2                 
[103] labeling_0.4.2                ps_1.7.5                     
[105] plyr_1.8.8                    getPass_0.2-2                
[107] fs_1.6.2                      stringi_1.7.12               
[109] viridisLite_0.4.2             BiocParallel_1.34.2          
[111] munsell_0.5.0                 Biostrings_2.68.1            
[113] lazyeval_0.2.2                V8_4.3.2                     
[115] Matrix_1.6-0                  hms_1.1.3                    
[117] bit64_4.0.5                   KEGGREST_1.40.0              
[119] shiny_1.7.4.1                 highr_0.10                   
[121] SummarizedExperiment_1.30.2   interactiveDisplayBase_1.38.0
[123] igraph_1.5.0                  memoise_2.0.1                
[125] bslib_0.5.0                   bit_4.0.5                    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
